<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>An S4 class to represent a set of PGS Catalog Polygenic Scores — scores-class • quincunx</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="An S4 class to represent a set of PGS Catalog Polygenic Scores — scores-class" />
<meta property="og:description" content="The scores object consists of six tables (slots) that combined form a
relational database of a subset of PGS Catalog polygenic scores. Each score
is an observation (row) in the scores table (the first table)." />
<meta property="og:image" content="https://rmagno.eu/quincunx/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">quincunx</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="divider"></li>
    <li class="dropdown-header">PGS Catalog Concepts</li>
    <li>
      <a href="../articles/identifiers.html">Identifiers</a>
    </li>
    <li>
      <a href="../articles/pgs-construction-process.html">PGS Construction Process</a>
    </li>
    <li>
      <a href="../articles/cohorts-samples-sample-sets.html">Cohorts, Samples and Sample Sets</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Getting PGS Catalog Data</li>
    <li>
      <a href="../articles/getting-pgs-scores.html">Getting PGS Scores</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-publications.html">Getting PGS Publications</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-sample-sets.html">Getting PGS Sample Sets</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-performance-metrics.html">Getting PGS Performance Metrics</a>
    </li>
    <li>
      <a href="../articles/getting-traits.html">Getting Traits</a>
    </li>
    <li>
      <a href="../articles/getting-cohorts.html">Getting Cohorts</a>
    </li>
    <li>
      <a href="../articles/getting-releases.html">PGS Catalog Releases</a>
    </li>
    <li>
      <a href="../articles/pgs-scoring-file.html">PGS scoring file</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Case Studies</li>
    <li>
      <a href="../articles/pgs-scores-mavaddat.html">PGS scores by Mavaddat et al. (2018)</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="../articles/index.html">More...</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
      <form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>An S4 class to represent a set of PGS Catalog Polygenic Scores</h1>
    
    <div class="hidden name"><code>scores-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The scores object consists of six tables (slots) that combined form a
relational database of a subset of PGS Catalog polygenic scores. Each score
is an observation (row) in the <code>scores</code> table (the first table).</p>
    </div>



    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    

<dl>
<dt><code>scores</code></dt><dd><p>A table of polygenic scores. Each polygenic score (row) is
  uniquely identified by the <code>pgs_id</code> column. Columns:
<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier. Example: <code>"PGS000001"</code>.</p></dd>
<dt>pgs_name</dt><dd><p>This may be the name that the authors describe the PGS with
in the source publication, or a name that a curator of the PGS Catalog has
assigned to identify the score during the curation process (before a PGS
identifier has been given). Example: <code>PRS77_BC</code>.</p></dd>
<dt>scoring_file</dt><dd><p>URL to the scoring file on the PGS FTP server. Example:
<code>"http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000001/ScoringFiles/PGS000001.txt.gz"</code>.</p></dd>
<dt>matches_publication</dt><dd><p>Indicate if the PGS data matches the published
polygenic score (<code>TRUE</code>). If not (<code>FALSE</code>), the authors have
provided an alternative polygenic for the Catalog and some other data, such
as performance metrics, may differ from the publication.</p></dd>
<dt>reported_trait</dt><dd><p>The author-reported trait that the PGS has been
developed to predict. Example: <code>"Breast Cancer"</code>.</p></dd>
<dt>trait_additional_description</dt><dd><p>Any additional description not captured
in the other columns. Example: <code>"Femoral neck BMD (g/cm2)"</code>.</p></dd>
<dt>pgs_method_name</dt><dd><p>The name or description of the method or computational
algorithm used to develop the PGS.</p></dd>
<dt>pgs_method_params</dt><dd><p>A description of the relevant inputs and parameters
relevant to the PGS development method/process.</p></dd>
<dt>n_variants</dt><dd><p>Number of variants used to calculate the PGS.</p></dd>
<dt>n_variants_interactions</dt><dd><p>Number of higher-order variant interactions
included in the PGS.</p></dd>
<dt>assembly</dt><dd><p>The version of the genome assembly that the variants present
in the PGS are associated with. Example: <code>GRCh37</code>.</p></dd>
<dt>license</dt><dd><p>The PGS Catalog distributes its data according to EBI’s
standard Terms of Use. Some PGS have specific terms, licenses, or
restrictions (e.g. non-commercial use) that we highlight in this field, if
known.</p></dd></dl></p></dd>

<dt><code>publications</code></dt><dd><p>A table of publications. Each publication (row) is
  uniquely identified by the <code>pgp_id</code> column. Columns:
<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p></dd>
<dt>pgp_id</dt><dd><p>PGS Publication identifier. Example: <code>"PGP000001"</code>.</p></dd>
<dt>pubmed_id</dt><dd><p><a href='https://en.wikipedia.org/wiki/PubMed'>PubMed</a>
identifier. Example: <code>"25855707"</code>.</p></dd>
<dt>publication_date</dt><dd><p>Publication date. Example: <code>"2020-09-28"</code>. Note
that the class of <code>publication_date</code> is <code><a href='https://rdrr.io/r/base/Dates.html'>Date</a></code>.</p></dd>
<dt>publication</dt><dd><p>Abbreviated name of the journal. Example: <code>"Am J Hum
Genet"</code>.</p></dd>
<dt>title</dt><dd><p>Publication title.</p></dd>
<dt>author_fullname</dt><dd><p>First author of the publication. Example:
<code>'Mavaddat N'</code>.</p></dd>
<dt>doi</dt><dd><p>Digital Object Identifier (DOI). This variable is also curated to
allow unpublished work (e.g. pre-prints) to be added to the catalog. Example:
<code>"10.1093/jnci/djv036"</code>.</p></dd></dl></p></dd>

<dt><code>samples</code></dt><dd><p>A table of samples. Each sample (row) is uniquely identified by
  the combination of values from the columns: <code>pgs_id</code> and
  <code>sample_id</code>. Columns:
<dl>
<dt>pgs_id</dt><dd><p>Polygenic score identifier. An identifier that starts with
<code>'PGS'</code> and is followed by six digits, e.g. <code>'PGS000001'</code>.</p></dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p></dd>
<dt>stage</dt><dd><p>Sample stage: either <code>"discovery"</code> or <code>"training"</code>.</p></dd>
<dt>sample_size</dt><dd><p>Number of individuals included in the sample.</p></dd>
<dt>sample_cases</dt><dd><p>Number of cases.</p></dd>
<dt>sample_controls</dt><dd><p>Number of controls.</p></dd>
<dt>sample_percent_male</dt><dd><p>Percentage of male participants.</p></dd>
<dt>phenotype_description</dt><dd><p>Detailed phenotype description.</p></dd>
<dt>ancestry</dt><dd><p>Author reported ancestry is mapped to the best matching
ancestry category from the NHGRI-EBI GWAS Catalog framework (see
<code><a href='ancestry_categories.html'>ancestry_categories</a></code>) for possible values.</p></dd>
<dt>ancestry_description</dt><dd><p>A more detailed description of sample ancestry
that usually matches the most specific description described by the authors
(e.g. French, Chinese).</p></dd>
<dt>ancestry_country</dt><dd><p>Author reported countries of recruitment (if available).</p></dd>
<dt>ancestry_additional_description</dt><dd><p>Any additional description not
captured in the other columns (e.g. founder or genetically isolated
populations, or further description of admixed samples).</p></dd>
<dt>study_id</dt><dd><p>Associated GWAS Catalog study accession identifier, e.g.,
<code>"GCST002735"</code>.</p></dd>
<dt>pubmed_id</dt><dd><p><a href='https://en.wikipedia.org/wiki/PubMed'>PubMed</a>
identifier.</p></dd>
<dt>cohorts_additional_description</dt><dd><p>Any additional description about the
samples (e.g. sub-cohort information).</p></dd></dl></p></dd>

<dt><code>demographics</code></dt><dd><p>A table of sample demographics' variables. Each
  demographics' variable (row) is uniquely identified by the combination of
  values from the columns: <code>pgs_id</code>, <code>sample_id</code> and
  <code>variable</code>. Columns:
<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p></dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate identifier to
identify each sample.</p></dd>
<dt>variable</dt><dd><p>Demographics variable. Following columns report about the
indicated variable.</p></dd>
<dt>estimate_type</dt><dd><p>Type of statistical estimate for variable.</p></dd>
<dt>estimate</dt><dd><p>The variable's statistical value.</p></dd>
<dt>unit</dt><dd><p>Unit of the variable.</p></dd>
<dt>variability_type</dt><dd><p>Measure of statistical dispersion for variable, e.g.
standard error (se) or standard deviation (sd).</p></dd>
<dt>variability</dt><dd><p>The value of the measure of dispersion.</p></dd>
<dt>interval_type</dt><dd><p>Type of statistical interval for variable: range, iqr
(interquartile), ci (confidence interval).</p></dd>
<dt>interval_lower</dt><dd><p>Interval lower bound.</p></dd>
<dt>interval_upper</dt><dd><p>Interval upper bound.</p></dd></dl></p></dd>

<dt><code>cohorts</code></dt><dd><p>A table of cohorts. Each cohort (row) is uniquely identified by
  the combination of values from the columns: <code>pgs_id</code>, <code>sample_id</code>
  and <code>cohort_symbol</code>. Columns:
<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p></dd>
<dt>sample_id</dt><dd><p>Sample identifier. This is a surrogate key to identify each sample.</p></dd>
<dt>cohort_symbol</dt><dd><p>Cohort symbol.</p></dd>
<dt>cohort_name</dt><dd><p>Cohort full name.</p></dd></dl></p></dd>

<dt><code>traits</code></dt><dd><p>A table of EFO traits. Each trait (row) is uniquely identified
  by the combination of the columns <code>pgs_id</code> and <code>efo_id</code>. Columns:
<dl>
<dt>pgs_id</dt><dd><p>Polygenic Score (PGS) identifier.</p></dd>
<dt>efo_id</dt><dd><p>An <a href='https://www.ebi.ac.uk/efo/'>EFO</a> identifier.</p></dd>
<dt>trait</dt><dd><p>Trait name.</p></dd>
<dt>description</dt><dd><p>Detailed description of the trait from EFO.</p></dd>
<dt>url</dt><dd><p>External link to the EFO entry.</p></dd></dl></p></dd>

</dl>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://rmagno.eu/'>Ramiro Magno</a>, <a href='https://iduarte.eu/'>Isabel Duarte</a>, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '6eb557c1e1320049dc93c778f9efe639',
    indexName: 'quincunx',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


