<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exploring PGS scores by Mavaddat et al. (2018) • quincunx</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Exploring PGS scores by Mavaddat et al. (2018)">
<meta property="og:description" content="quincunx">
<meta property="og:image" content="https://maialab.org/quincunx/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">quincunx</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    <li class="dropdown-header">PGS Catalog Concepts</li>
    <li>
      <a href="../articles/identifiers.html">Identifiers</a>
    </li>
    <li>
      <a href="../articles/pgs-construction-process.html">PGS Construction Process</a>
    </li>
    <li>
      <a href="../articles/cohorts-samples-sample-sets.html">Cohorts, Samples and Sample Sets</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Getting PGS Catalog Data</li>
    <li>
      <a href="../articles/getting-pgs-scores.html">Getting PGS Scores</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-publications.html">Getting PGS Publications</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-sample-sets.html">Getting PGS Sample Sets</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-performance-metrics.html">Getting PGS Performance Metrics</a>
    </li>
    <li>
      <a href="../articles/getting-traits.html">Getting Traits</a>
    </li>
    <li>
      <a href="../articles/getting-cohorts.html">Getting Cohorts</a>
    </li>
    <li>
      <a href="../articles/getting-releases.html">PGS Catalog Releases</a>
    </li>
    <li>
      <a href="../articles/pgs-scoring-file.html">PGS scoring file</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Case Studies</li>
    <li>
      <a href="../articles/pgs-scores-mavaddat.html">Exploring PGS scores by Mavaddat et al. (2018)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/index.html">More...</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pgs-scores-mavaddat_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Exploring PGS scores by Mavaddat et al. (2018)</h1>
            
      
      
      <div class="hidden name"><code>pgs-scores-mavaddat.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>Let us assume that you are a breast cancer researcher, and that you are interested in studying the screening and prevention of this disease. Now, imagine you have just recently noticed a new publication that claims to have developed a set of new polygenic risk scores that are both powerful and reliable predictors of breast cancer risk, e.g., <a href="https://doi.org/10.1016/j.ajhg.2018.11.002">Mavaddat et al. (2018)</a>.</p>
<p>Perhaps now you would like to investigate a bit more about these new polygenic risk scores to assess their potential application. You know that the performance of these scores is dependent on various aspects, such as study design, participant demographics, case definitions, and covariates that have been adjusted for. In general, to access this information, you will have to carefully read the paper searching for these details, and usually get them from the supplementary material, with all the extra effort it takes.<br>
However, if these scores have already been indexed and manually curated by the <a href="https://www.pgscatalog.org/">PGS Catalog</a> team, then you can benefit by using this free and open resource to quickly gather relevant data about these scores. And if you happen to be an R user, then you can use the R Package <em>quincunx</em> and its retrieval functions to fetch the polygenic score metadata associated with the publication of interest from the PGS Catalog REST API server. This is what we will be doing next.</p>
</div>
<div id="finding-polygenic-risk-scores-by-publication" class="section level2">
<h2 class="hasAnchor">
<a href="#finding-polygenic-risk-scores-by-publication" class="anchor"></a>Finding polygenic risk scores by publication</h2>
<div id="searching-for-publications-by-author-name" class="section level3">
<h3 class="hasAnchor">
<a href="#searching-for-publications-by-author-name" class="anchor"></a>Searching for publications by author name</h3>
<p>To check if we are fortunate enough to have the polygenic risk scores for breast cancer (<a href="https://doi.org/10.1016/j.ajhg.2018.11.002">Mavaddat et al. (2018)</a>) in the PGS Catalog, we start by loading the quincunx package and by searching for this publication in the PGS Catalog. To do this we will use the function <code><a href="../reference/get_publications.html">get_publications()</a></code> and start by searching by the first author’s last name, i.e. “Mavaddat”:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">quincunx</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>

<span class="va">pub_by_mavaddat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_publications.html">get_publications</a></span><span class="op">(</span>author <span class="op">=</span> <span class="st">'Mavaddat'</span><span class="op">)</span> <span class="co"># Not case sensitive,</span>
                                                         <span class="co"># so "mavaddat" would</span>
                                                         <span class="co"># have worked just fine.</span></code></pre></div>
<p>When you ask the PGS Catalog for publications, it returns an S4 <code>publications</code> object with two tables (slots): <code>publications</code> and <code>pgs_ids</code>. You can access these tables with the <code><a href="https://rdrr.io/r/base/slotOp.html">@</a></code> operator.</p>
<p>The <code>publications</code> table (data frame) contains all the publications added to the PGS Catalog that contain an author whose last name is “Mavaddat”:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pub_by_mavaddat</span><span class="op">@</span><span class="va">publications</span>
<span class="co">#&gt; # A tibble: 6 x 8</span>
<span class="co">#&gt;   pgp_id pubmed_id publication_date publication title author_fullname doi  </span>
<span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;           &lt;chr&gt;</span>
<span class="co">#&gt; 1 PGP00… 25855707  2015-04-08       J Natl Can… Pred… Mavaddat N      10.1…</span>
<span class="co">#&gt; 2 PGP00… 30554720  2018-12-13       Am J Hum G… Poly… Mavaddat N      10.1…</span>
<span class="co">#&gt; 3 PGP00… 28376175  2017-07-01       J Natl Can… Eval… Kuchenbaecker … 10.1…</span>
<span class="co">#&gt; 4 PGP00… 33022221  2020-10-05       Am J Hum G… Brea… Kramer I        10.1…</span>
<span class="co">#&gt; 5 PGP00… 32737321  2020-07-31       Nat Commun  Euro… Ho WK           10.1…</span>
<span class="co">#&gt; 6 PGP00… 32665703  2020-07-15       Genet Med   Poly… Barnes DR       10.1…</span>
<span class="co">#&gt; # … with 1 more variable: authors &lt;chr&gt;</span></code></pre></div>
<p>Seemingly there are 6 publications indexed in the PGS Catalog. Each publication added to the PGS Catalog gets an unique identifier. This identifier is in the first column (<code>pgp_id</code>) of the <code>publications</code> table. The number of publications can be obtained by using the function <code><a href="../reference/n.html">n()</a></code> on the S4 object <code>pub_by_mavaddat</code>, or by asking directly the number of rows of the <code>publications</code> table:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">pub_by_mavaddat</span><span class="op">)</span> <span class="co"># Here we use quincunx::n() instead of n() to avoid namespace collisions with dplyr::n().</span>
<span class="co">#&gt; [1] 6</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">pub_by_mavaddat</span><span class="op">@</span><span class="va">publications</span><span class="op">)</span>
<span class="co">#&gt; [1] 6</span></code></pre></div>
<p>Moreover, we can see that Nasim Mavaddat is not the first author in all these publications because her name is not always present in the column <code>author_fullname</code> (which contains the name of the first author only). If you want to know all the authors you can access the column <code>authors</code> (contains a string of comma separated author names).</p>
<p>Now, to identify our publication of interest amongst these 6 publications, we can check, for example, the PubMed identifier of <a href="https://doi.org/10.1016/j.ajhg.2018.11.002">Mavaddat et al. (2018)</a> (column <code>pubmed_id</code>), or check other information such as journal name (column <code>publication</code>), the publication title (column <code>title</code>) or the publication date (column <code>publication_date</code>). The most unambiguous approach is to use the PubMed identifier. The PubMed identifier for <a href="https://doi.org/10.1016/j.ajhg.2018.11.002">Mavaddat et al. (2018)</a> is <code>"30554720"</code>. So the publication of interest corresponds to the one returned in row number 2 of the <code>publications</code> table:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">publication_row_index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">pub_by_mavaddat</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">pubmed_id</span> <span class="op">==</span> <span class="st">"30554720"</span><span class="op">)</span>
<span class="va">publication_row_index</span>
<span class="co">#&gt; [1] 2</span></code></pre></div>
<p>To focus now on this publication and filter out the others, we can subset the <code>publications</code> object <code>pub_by_mavaddat</code>, e.g., by position, and get a new object with both tables (<code>publications</code> and <code>pgs_ids</code>) filtered for only this publication:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018</span> <span class="op">&lt;-</span> <span class="va">pub_by_mavaddat</span><span class="op">[</span><span class="va">publication_row_index</span><span class="op">]</span>
<span class="va">mavaddat2018</span>
<span class="co">#&gt; An object of class "publications"</span>
<span class="co">#&gt; Slot "publications":</span>
<span class="co">#&gt; # A tibble: 1 x 8</span>
<span class="co">#&gt;   pgp_id pubmed_id publication_date publication title author_fullname doi  </span>
<span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;           &lt;chr&gt;</span>
<span class="co">#&gt; 1 PGP00… 30554720  2018-12-13       Am J Hum G… Poly… Mavaddat N      10.1…</span>
<span class="co">#&gt; # … with 1 more variable: authors &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "pgs_ids":</span>
<span class="co">#&gt; # A tibble: 15 x 3</span>
<span class="co">#&gt;    pgp_id    pgs_id    stage      </span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt;  1 PGP000002 PGS000004 development</span>
<span class="co">#&gt;  2 PGP000002 PGS000005 development</span>
<span class="co">#&gt;  3 PGP000002 PGS000006 development</span>
<span class="co">#&gt;  4 PGP000002 PGS000007 development</span>
<span class="co">#&gt;  5 PGP000002 PGS000008 development</span>
<span class="co">#&gt;  6 PGP000002 PGS000009 development</span>
<span class="co">#&gt;  7 PGP000002 PGS000001 evaluation </span>
<span class="co">#&gt;  8 PGP000002 PGS000002 evaluation </span>
<span class="co">#&gt;  9 PGP000002 PGS000003 evaluation </span>
<span class="co">#&gt; 10 PGP000002 PGS000004 evaluation </span>
<span class="co">#&gt; 11 PGP000002 PGS000005 evaluation </span>
<span class="co">#&gt; 12 PGP000002 PGS000006 evaluation </span>
<span class="co">#&gt; 13 PGP000002 PGS000007 evaluation </span>
<span class="co">#&gt; 14 PGP000002 PGS000008 evaluation </span>
<span class="co">#&gt; 15 PGP000002 PGS000009 evaluation</span></code></pre></div>
<p>We see now that the <code>publications</code> table contains one entry only, and that the PGS Catalog identifier assigned to this publication is PGP000002:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">pgp_id</span>
<span class="co">#&gt; [1] "PGP000002"</span></code></pre></div>
<p>In the <code>pgs_ids</code> we see that there are several PGS scores associated with publication PGP000002. Besides listing the score identifiers (<code>pgs_id</code>), it also includes the <code>stage</code> column that annotates the polygenic score relative to the PGS construction process (check page 2 of the <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf">quincunx cheatsheet</a> for a visual illustration).</p>
</div>
<div id="searching-for-publications-by-pubmed-id" class="section level3">
<h3 class="hasAnchor">
<a href="#searching-for-publications-by-pubmed-id" class="anchor"></a>Searching for publications by PubMed ID</h3>
<p>An alternatively, albeit more direct, route to get this publication by <a href="https://doi.org/10.1016/j.ajhg.2018.11.002">Mavaddat et al. (2018)</a> could have been to query for publications directly by the corresponding PubMed ID (30554720):</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pub_by_pmid_30554720</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_publications.html">get_publications</a></span><span class="op">(</span>pubmed_id <span class="op">=</span> <span class="st">'30554720'</span><span class="op">)</span>
<span class="va">pub_by_pmid_30554720</span>
<span class="co">#&gt; An object of class "publications"</span>
<span class="co">#&gt; Slot "publications":</span>
<span class="co">#&gt; # A tibble: 1 x 8</span>
<span class="co">#&gt;   pgp_id pubmed_id publication_date publication title author_fullname doi  </span>
<span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;           &lt;chr&gt;</span>
<span class="co">#&gt; 1 PGP00… 30554720  2018-12-13       Am J Hum G… Poly… Mavaddat N      10.1…</span>
<span class="co">#&gt; # … with 1 more variable: authors &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slot "pgs_ids":</span>
<span class="co">#&gt; # A tibble: 15 x 3</span>
<span class="co">#&gt;    pgp_id    pgs_id    stage      </span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt;  1 PGP000002 PGS000004 development</span>
<span class="co">#&gt;  2 PGP000002 PGS000005 development</span>
<span class="co">#&gt;  3 PGP000002 PGS000006 development</span>
<span class="co">#&gt;  4 PGP000002 PGS000007 development</span>
<span class="co">#&gt;  5 PGP000002 PGS000008 development</span>
<span class="co">#&gt;  6 PGP000002 PGS000009 development</span>
<span class="co">#&gt;  7 PGP000002 PGS000001 evaluation </span>
<span class="co">#&gt;  8 PGP000002 PGS000002 evaluation </span>
<span class="co">#&gt;  9 PGP000002 PGS000003 evaluation </span>
<span class="co">#&gt; 10 PGP000002 PGS000004 evaluation </span>
<span class="co">#&gt; 11 PGP000002 PGS000005 evaluation </span>
<span class="co">#&gt; 12 PGP000002 PGS000006 evaluation </span>
<span class="co">#&gt; 13 PGP000002 PGS000007 evaluation </span>
<span class="co">#&gt; 14 PGP000002 PGS000008 evaluation </span>
<span class="co">#&gt; 15 PGP000002 PGS000009 evaluation</span></code></pre></div>
<p>To get an overview of the possible search criteria for <code>get_publications</code> you can use the help function within R.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">get_publications</span>

<span class="co"># or alternatively</span>
<span class="fu"><a href="https://rdrr.io/r/utils/help.html">help</a></span><span class="op">(</span><span class="st">"get_publications"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="from-publication-to-polygenic-risk-scores" class="section level2">
<h2 class="hasAnchor">
<a href="#from-publication-to-polygenic-risk-scores" class="anchor"></a>From publication to polygenic risk scores</h2>
<p>Now that we have found that our publication of interest exists in the PGS Catalog, with identifier PGP000002, we can check now which polygenic risk scores are annotated in the Catalog. Polygenic scores (PGS) in the PGS Catalog are indexed by an unique accession identifier of the form: “PGS000000” (“PGS” followed by six digits).</p>
<p>To get all PGS identifiers associated with Mavaddat’s publication we turn to the second table <code>pgs_ids</code> that maps publication identifiers (PGP) to PGS identifiers:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pub_by_pmid_30554720</span><span class="op">@</span><span class="va">pgs_ids</span>
<span class="co">#&gt; # A tibble: 15 x 3</span>
<span class="co">#&gt;    pgp_id    pgs_id    stage      </span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt;  1 PGP000002 PGS000004 development</span>
<span class="co">#&gt;  2 PGP000002 PGS000005 development</span>
<span class="co">#&gt;  3 PGP000002 PGS000006 development</span>
<span class="co">#&gt;  4 PGP000002 PGS000007 development</span>
<span class="co">#&gt;  5 PGP000002 PGS000008 development</span>
<span class="co">#&gt;  6 PGP000002 PGS000009 development</span>
<span class="co">#&gt;  7 PGP000002 PGS000001 evaluation </span>
<span class="co">#&gt;  8 PGP000002 PGS000002 evaluation </span>
<span class="co">#&gt;  9 PGP000002 PGS000003 evaluation </span>
<span class="co">#&gt; 10 PGP000002 PGS000004 evaluation </span>
<span class="co">#&gt; 11 PGP000002 PGS000005 evaluation </span>
<span class="co">#&gt; 12 PGP000002 PGS000006 evaluation </span>
<span class="co">#&gt; 13 PGP000002 PGS000007 evaluation </span>
<span class="co">#&gt; 14 PGP000002 PGS000008 evaluation </span>
<span class="co">#&gt; 15 PGP000002 PGS000009 evaluation</span></code></pre></div>
<p>Please note that there are 9 unique scores, both from the <em>development</em> and the <em>evaluation</em> stages, meaning that this paper published new polygenic scores (<em>development</em> stage), and tested them (<em>evaluation</em> stage). But this paper has also evaluated 3 other polygenic scores previously developed (and firstly published in another publication by the same author).<br><strong>This distinction between stages is important because when we query the database for the scores from all the pgp_ids present in this publication, only the newly developed ones (from the <em>development stage</em>) will be retrieved.</strong> (See below: section about the <code><a href="../reference/get_scores.html">get_scores()</a></code> function).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Newly published PGS scores (development stage)</span>
<span class="va">pub_by_pmid_30554720</span><span class="op">@</span><span class="va">pgs_ids</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"development"</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 3</span>
<span class="co">#&gt;   pgp_id    pgs_id    stage      </span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt; 1 PGP000002 PGS000004 development</span>
<span class="co">#&gt; 2 PGP000002 PGS000005 development</span>
<span class="co">#&gt; 3 PGP000002 PGS000006 development</span>
<span class="co">#&gt; 4 PGP000002 PGS000007 development</span>
<span class="co">#&gt; 5 PGP000002 PGS000008 development</span>
<span class="co">#&gt; 6 PGP000002 PGS000009 development</span>

<span class="co"># All PGS scores evaluated (evaluation stage)  </span>
<span class="va">pub_by_pmid_30554720</span><span class="op">@</span><span class="va">pgs_ids</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"evaluation"</span><span class="op">)</span> 
<span class="co">#&gt; # A tibble: 9 x 3</span>
<span class="co">#&gt;   pgp_id    pgs_id    stage     </span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;     </span>
<span class="co">#&gt; 1 PGP000002 PGS000001 evaluation</span>
<span class="co">#&gt; 2 PGP000002 PGS000002 evaluation</span>
<span class="co">#&gt; 3 PGP000002 PGS000003 evaluation</span>
<span class="co">#&gt; 4 PGP000002 PGS000004 evaluation</span>
<span class="co">#&gt; 5 PGP000002 PGS000005 evaluation</span>
<span class="co">#&gt; 6 PGP000002 PGS000006 evaluation</span>
<span class="co">#&gt; 7 PGP000002 PGS000007 evaluation</span>
<span class="co">#&gt; 8 PGP000002 PGS000008 evaluation</span>
<span class="co">#&gt; 9 PGP000002 PGS000009 evaluation</span></code></pre></div>
<p>If we knew, before hand, that PGP000002 was associated with Mavaddat’s publication, we could have also taken advantage of the neat function <code><a href="../reference/pgp_to_pgs.html">pgp_to_pgs()</a></code> to quickly get all the associated polygenic score ids:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/pgp_to_pgs.html">pgp_to_pgs</a></span><span class="op">(</span><span class="st">'PGP000002'</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 15 x 3</span>
<span class="co">#&gt;    pgp_id    pgs_id    stage      </span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt;  1 PGP000002 PGS000004 development</span>
<span class="co">#&gt;  2 PGP000002 PGS000005 development</span>
<span class="co">#&gt;  3 PGP000002 PGS000006 development</span>
<span class="co">#&gt;  4 PGP000002 PGS000007 development</span>
<span class="co">#&gt;  5 PGP000002 PGS000008 development</span>
<span class="co">#&gt;  6 PGP000002 PGS000009 development</span>
<span class="co">#&gt;  7 PGP000002 PGS000001 evaluation </span>
<span class="co">#&gt;  8 PGP000002 PGS000002 evaluation </span>
<span class="co">#&gt;  9 PGP000002 PGS000003 evaluation </span>
<span class="co">#&gt; 10 PGP000002 PGS000004 evaluation </span>
<span class="co">#&gt; 11 PGP000002 PGS000005 evaluation </span>
<span class="co">#&gt; 12 PGP000002 PGS000006 evaluation </span>
<span class="co">#&gt; 13 PGP000002 PGS000007 evaluation </span>
<span class="co">#&gt; 14 PGP000002 PGS000008 evaluation </span>
<span class="co">#&gt; 15 PGP000002 PGS000009 evaluation</span></code></pre></div>
</div>
<div id="polygenic-scores-published-by-mavaddat-et-al--2018" class="section level2">
<h2 class="hasAnchor">
<a href="#polygenic-scores-published-by-mavaddat-et-al--2018" class="anchor"></a>Polygenic Scores published by Mavaddat <em>et al.</em> (2018)</h2>
<p>To dive into the metadata about these polygenic scores, we use the quincunx function <code><a href="../reference/get_scores.html">get_scores()</a></code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_scores.html">get_scores</a></span><span class="op">(</span>pubmed_id <span class="op">=</span> <span class="st">"30554720"</span><span class="op">)</span>
<span class="fu">slotNames</span><span class="op">(</span><span class="va">mavaddat2018_scores</span><span class="op">)</span>
<span class="co">#&gt; [1] "scores"       "publications" "samples"      "demographics" "cohorts"     </span>
<span class="co">#&gt; [6] "traits"</span></code></pre></div>
<p>This returns the S4 object <code>scores</code> which contains six tables (slots): scores, publications, samples, demographics, cohorts, traits.<br>
You can quickly check all variables from each table by consulting <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf">quincunx cheatsheet</a>.<br>
A description of each variable is annotated in the <code>scores</code> object help page that can be accessed with <code><a href="../reference/scores-class.html">class?scores</a></code>.</p>
</div>
<div id="polygenic-scores-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#polygenic-scores-metadata" class="anchor"></a>Polygenic Scores Metadata</h2>
<div id="scores-object-scores-table" class="section level3">
<h3 class="hasAnchor">
<a href="#scores-object-scores-table" class="anchor"></a>scores object | scores table</h3>
<p>The S4 scores object <code>mavaddat2018_scores</code> starts with a table named <code>scores</code> that lists each score in one row. All scores are identified by a PGS identifier (<code>pgs_id</code> column). Note that, as previously explained, only the polygenic scores newly developed in this publication (<em>developmental stage</em>) are retrieved, and not all the PGS scores that were evaluated in this publication.</p>
<p>In addition, scores can have a name (<code>pgs_name</code> column). This may be the name assigned by authors of the source publication, or a name assigned by a PGS Catalog curator in order to identify that particular score throughout the curating process.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'pgs_name'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   pgs_id    pgs_name     </span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;        </span>
<span class="co">#&gt; 1 PGS000004 PRS313_BC    </span>
<span class="co">#&gt; 2 PGS000005 PRS313_ERpos </span>
<span class="co">#&gt; 3 PGS000006 PRS313_ERneg </span>
<span class="co">#&gt; 4 PGS000007 PRS3820_BC   </span>
<span class="co">#&gt; 5 PGS000008 PRS3820_ERpos</span>
<span class="co">#&gt; 6 PGS000009 PRS3820_ERneg</span></code></pre></div>
<p>From the PGS names we can already see the presence of the suffixes “ERpos” and “ERneg”, suggestive of specialized polygenic risk scores for estrogen-receptor positive and negative samples.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="st">'scoring_file'</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 6 x 1</span>
<span class="co">#&gt;   scoring_file                                                                  </span>
<span class="co">#&gt;   &lt;chr&gt;                                                                         </span>
<span class="co">#&gt; 1 http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000004/ScoringFiles/PGS…</span>
<span class="co">#&gt; 2 http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000005/ScoringFiles/PGS…</span>
<span class="co">#&gt; 3 http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000006/ScoringFiles/PGS…</span>
<span class="co">#&gt; 4 http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000007/ScoringFiles/PGS…</span>
<span class="co">#&gt; 5 http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000008/ScoringFiles/PGS…</span>
<span class="co">#&gt; 6 http://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000009/ScoringFiles/PGS…</span></code></pre></div>
<p>The column <code>scoring_file</code> contains the URL for the FTP location containing the corresponding PGS scoring files. PGS scoring files are the text files provided by the PGS Catalog team containing the source data that you can use to compute polygenic scores for particular individuals, i.e. that allow you to apply these scores to your individual samples. Learn more about scoring files in <code><a href="../articles/pgs-scoring-file.html">vignette("pgs-scoring-file")</a></code>. For a quick consultation of the file format of PGS scoring files you may also check the second page of <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf">quincunx cheatsheet</a>.</p>
<p>As an additional feature, <em>quincunx</em> allows you to download the relevant PGS scoring files directly into R using the function <code><a href="../reference/read_scoring_file.html">read_scoring_file()</a></code>, making the data immediately available in R for further analysis.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'matches_publication'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   pgs_id    matches_publication</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;lgl&gt;              </span>
<span class="co">#&gt; 1 PGS000004 TRUE               </span>
<span class="co">#&gt; 2 PGS000005 TRUE               </span>
<span class="co">#&gt; 3 PGS000006 TRUE               </span>
<span class="co">#&gt; 4 PGS000007 TRUE               </span>
<span class="co">#&gt; 5 PGS000008 TRUE               </span>
<span class="co">#&gt; 6 PGS000009 TRUE</span></code></pre></div>
<p>The column <code>matches_publication</code> is a logical value indicating whether the published polygenic score is exactly the same as the one present in the PGS scoring file provided by the PGS Catalog. In this case all of the 6 scores are provided exactly as published (all values are “TRUE”).</p>
<p>Other columns in the <code>scores</code> table hold relevant information.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_name'</span>, <span class="st">'pgs_method_name'</span>, <span class="st">'pgs_method_params'</span>, <span class="st">'n_variants'</span>, <span class="st">'assembly'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 6 x 5</span>
<span class="co">#&gt;   pgs_name    pgs_method_name               pgs_method_para… n_variants assembly</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;                         &lt;chr&gt;                 &lt;int&gt; &lt;chr&gt;   </span>
<span class="co">#&gt; 1 PRS313_BC   Hard-Thresholding Stepwise F… p &lt; 10^-5               313 GRCh37  </span>
<span class="co">#&gt; 2 PRS313_ERp… Hard-Thresholding Stepwise F… p &lt; 10^-5               313 GRCh37  </span>
<span class="co">#&gt; 3 PRS313_ERn… Hard-Thresholding Stepwise F… p &lt; 10^-5               313 GRCh37  </span>
<span class="co">#&gt; 4 PRS3820_BC  LASSO penalized regression    p &lt; 0.001              3820 GRCh37  </span>
<span class="co">#&gt; 5 PRS3820_ER… LASSO penalized regression    p &lt; 0.001              3820 GRCh37  </span>
<span class="co">#&gt; 6 PRS3820_ER… LASSO penalized regression    p &lt; 0.001              3820 GRCh37</span></code></pre></div>
<p>For example, columns such as <code>pgs_method_name</code> and <code>pgs_method_params</code> provide extra details about the PGS development method. Finally, <code>n_variants</code> informs about the number of variants comprising each polygenic risk score, and <code>assembly</code> indicates the genome assembly version used.</p>
</div>
<div id="scores-object-publications-table" class="section level3">
<h3 class="hasAnchor">
<a href="#scores-object-publications-table" class="anchor"></a>scores object | publications table</h3>
<p>The <code>scores</code> S4 object contains a table dedicated to the source publications used to collect the score(s) retrieved. In this case, it is not surprising that all PGS scores map to the same publication identifier, i.e., PGP000002, as that was our starting point.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">publications</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'pgp_id'</span>, <span class="st">'publication_date'</span>, <span class="st">'author_fullname'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 6 x 4</span>
<span class="co">#&gt;   pgs_id    pgp_id    publication_date author_fullname</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;     &lt;date&gt;           &lt;chr&gt;          </span>
<span class="co">#&gt; 1 PGS000004 PGP000002 2018-12-13       Mavaddat N     </span>
<span class="co">#&gt; 2 PGS000005 PGP000002 2018-12-13       Mavaddat N     </span>
<span class="co">#&gt; 3 PGS000006 PGP000002 2018-12-13       Mavaddat N     </span>
<span class="co">#&gt; 4 PGS000007 PGP000002 2018-12-13       Mavaddat N     </span>
<span class="co">#&gt; 5 PGS000008 PGP000002 2018-12-13       Mavaddat N     </span>
<span class="co">#&gt; 6 PGS000009 PGP000002 2018-12-13       Mavaddat N</span></code></pre></div>
</div>
<div id="scores-object-samples-table" class="section level3">
<h3 class="hasAnchor">
<a href="#scores-object-samples-table" class="anchor"></a>scores object | samples table</h3>
<p>The third table (slot) of the <code>scores</code> S4 object pertains to the samples used for the development of the PGS scores.</p>
<p>There are a total of 15 columns with metadata details about each sample. Each row corresponds to one sample associated with the polygenic scores, and the combination of values of the first two columns, <code>pgs_id</code> and <code>sample_id</code>, uniquely identifies each sample in this table. All samples shown in the <code>samples</code> table of a <code>scores</code> object are annotated with a <code>stage</code>, that can take two values: <code>"discovery"</code> or <code>"training"</code>.<br>
The <code>"discovery"</code> samples are typically used in determining the variants that are afterwards used for polygenic score development. These variants originate typically from Genome-Wide Association Studies (GWAS). Hence, these samples might be linked to the GWAS Catalog. If that is the case, this information is provided in the column <code>study_id</code>, indicating the GWAS Catalog accession identifier. You may find more information about these GWAS studies by using the function <code><a href="https://rdrr.io/pkg/gwasrapidd/man/get_studies.html">gwasrapidd::get_studies()</a></code> from the <a href="https://rmagno.eu/gwasrapidd/">gwasrapidd</a> R package that we developed previously (<a href="https://doi.org/10.1093/bioinformatics/btz605">described here</a>).<br>
The <code>"training"</code> samples are those that have been used for the training of a particular polygenic score. Together, these two stages (<em>discovery</em> and <em>training</em>) are referred to as <em>development</em>, in contrast to the later testing phase of the polygenic scores, i.e., the <em>evaluation</em> phase (or stage).<br>
If this sounds confusing check our <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf">cheatsheet</a>, section <em>PGS Construction Process</em>, second page.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">)</span>
<span class="co">#&gt; Rows: 18</span>
<span class="co">#&gt; Columns: 15</span>
<span class="co">#&gt; $ pgs_id                          &lt;chr&gt; "PGS000004", "PGS000004", "PGS000004"…</span>
<span class="co">#&gt; $ sample_id                       &lt;int&gt; 1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3, 1…</span>
<span class="co">#&gt; $ stage                           &lt;chr&gt; "discovery", "training", "training", …</span>
<span class="co">#&gt; $ sample_size                     &lt;int&gt; 139274, 158648, 10444, 139274, 87368,…</span>
<span class="co">#&gt; $ sample_cases                    &lt;int&gt; NA, 88916, 5159, NA, 55391, 4233, NA,…</span>
<span class="co">#&gt; $ sample_controls                 &lt;int&gt; NA, 69732, 5285, NA, 31977, 926, NA, …</span>
<span class="co">#&gt; $ sample_percent_male             &lt;dbl&gt; NA, 0, 0, NA, 0, 0, NA, 0, 0, NA, 0, …</span>
<span class="co">#&gt; $ phenotype_description           &lt;chr&gt; NA, "Invasive breast cancer-affected"…</span>
<span class="co">#&gt; $ ancestry                        &lt;chr&gt; "European", "European", "European", "…</span>
<span class="co">#&gt; $ ancestry_description            &lt;chr&gt; "NR", NA, NA, "NR", NA, NA, "NR", NA,…</span>
<span class="co">#&gt; $ ancestry_country                &lt;chr&gt; "Canada, U.S., Australia, Belgium, Fr…</span>
<span class="co">#&gt; $ ancestry_additional_description &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</span>
<span class="co">#&gt; $ study_id                        &lt;chr&gt; "GCST004988", NA, NA, "GCST004988", N…</span>
<span class="co">#&gt; $ pubmed_id                       &lt;chr&gt; "29059683", NA, NA, "29059683", NA, N…</span>
<span class="co">#&gt; $ cohorts_additional_description  &lt;chr&gt; NA, "Training Cohort", "Validation Co…</span></code></pre></div>
<p>The PGS Catalog provides brief records of samples sizes (total number, number of cases, and number of controls):</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 18 x 6</span>
<span class="co">#&gt;    pgs_id    sample_id stage     sample_size sample_cases sample_controls</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;           &lt;int&gt;        &lt;int&gt;           &lt;int&gt;</span>
<span class="co">#&gt;  1 PGS000004         1 discovery      139274           NA              NA</span>
<span class="co">#&gt;  2 PGS000004         2 training       158648        88916           69732</span>
<span class="co">#&gt;  3 PGS000004         3 training        10444         5159            5285</span>
<span class="co">#&gt;  4 PGS000005         1 discovery      139274           NA              NA</span>
<span class="co">#&gt;  5 PGS000005         2 training        87368        55391           31977</span>
<span class="co">#&gt;  6 PGS000005         3 training         5159         4233             926</span>
<span class="co">#&gt;  7 PGS000006         1 discovery      139274           NA              NA</span>
<span class="co">#&gt;  8 PGS000006         2 training        87368        15404           71964</span>
<span class="co">#&gt;  9 PGS000006         3 training         5159          926            4233</span>
<span class="co">#&gt; 10 PGS000007         1 discovery      139274           NA              NA</span>
<span class="co">#&gt; 11 PGS000007         2 training       158648        88916           69732</span>
<span class="co">#&gt; 12 PGS000007         3 training        10444         5159            5285</span>
<span class="co">#&gt; 13 PGS000008         1 discovery      139274           NA              NA</span>
<span class="co">#&gt; 14 PGS000008         2 training        87368        55391           31977</span>
<span class="co">#&gt; 15 PGS000008         3 training         5159         4233             926</span>
<span class="co">#&gt; 16 PGS000009         1 discovery      139274           NA              NA</span>
<span class="co">#&gt; 17 PGS000009         2 training        87368        15404           71964</span>
<span class="co">#&gt; 18 PGS000009         3 training         5159          926            4233</span></code></pre></div>
<p>Perhaps, not so surprisingly, the percentage of male participants in the training samples is zero:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'sample_id'</span>, <span class="st">'stage'</span>, <span class="st">'sample_percent_male'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 18 x 4</span>
<span class="co">#&gt;    pgs_id    sample_id stage     sample_percent_male</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;                   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 PGS000004         1 discovery                  NA</span>
<span class="co">#&gt;  2 PGS000004         2 training                    0</span>
<span class="co">#&gt;  3 PGS000004         3 training                    0</span>
<span class="co">#&gt;  4 PGS000005         1 discovery                  NA</span>
<span class="co">#&gt;  5 PGS000005         2 training                    0</span>
<span class="co">#&gt;  6 PGS000005         3 training                    0</span>
<span class="co">#&gt;  7 PGS000006         1 discovery                  NA</span>
<span class="co">#&gt;  8 PGS000006         2 training                    0</span>
<span class="co">#&gt;  9 PGS000006         3 training                    0</span>
<span class="co">#&gt; 10 PGS000007         1 discovery                  NA</span>
<span class="co">#&gt; 11 PGS000007         2 training                    0</span>
<span class="co">#&gt; 12 PGS000007         3 training                    0</span>
<span class="co">#&gt; 13 PGS000008         1 discovery                  NA</span>
<span class="co">#&gt; 14 PGS000008         2 training                    0</span>
<span class="co">#&gt; 15 PGS000008         3 training                    0</span>
<span class="co">#&gt; 16 PGS000009         1 discovery                  NA</span>
<span class="co">#&gt; 17 PGS000009         2 training                    0</span>
<span class="co">#&gt; 18 PGS000009         3 training                    0</span></code></pre></div>
<p>Also, information about the trait or disease studied and ancestry information can be accessed:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'sample_id'</span>, <span class="st">'stage'</span>, <span class="st">'phenotype_description'</span>, <span class="st">'ancestry'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 18 x 5</span>
<span class="co">#&gt;    pgs_id    sample_id stage     phenotype_description           ancestry</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;                           &lt;chr&gt;   </span>
<span class="co">#&gt;  1 PGS000004         1 discovery &lt;NA&gt;                            European</span>
<span class="co">#&gt;  2 PGS000004         2 training  Invasive breast cancer-affected European</span>
<span class="co">#&gt;  3 PGS000004         3 training  Invasive breast cancer-affected European</span>
<span class="co">#&gt;  4 PGS000005         1 discovery &lt;NA&gt;                            European</span>
<span class="co">#&gt;  5 PGS000005         2 training  ER-positive breast cancer cases European</span>
<span class="co">#&gt;  6 PGS000005         3 training  ER-positive breast cancer cases European</span>
<span class="co">#&gt;  7 PGS000006         1 discovery &lt;NA&gt;                            European</span>
<span class="co">#&gt;  8 PGS000006         2 training  ER-negative breast cancer cases European</span>
<span class="co">#&gt;  9 PGS000006         3 training  ER-negative breast cancer cases European</span>
<span class="co">#&gt; 10 PGS000007         1 discovery &lt;NA&gt;                            European</span>
<span class="co">#&gt; 11 PGS000007         2 training  Invasive breast cancer-affected European</span>
<span class="co">#&gt; 12 PGS000007         3 training  Invasive breast cancer-affected European</span>
<span class="co">#&gt; 13 PGS000008         1 discovery &lt;NA&gt;                            European</span>
<span class="co">#&gt; 14 PGS000008         2 training  ER-positive breast cancer cases European</span>
<span class="co">#&gt; 15 PGS000008         3 training  ER-positive breast cancer cases European</span>
<span class="co">#&gt; 16 PGS000009         1 discovery &lt;NA&gt;                            European</span>
<span class="co">#&gt; 17 PGS000009         2 training  ER-negative breast cancer cases European</span>
<span class="co">#&gt; 18 PGS000009         3 training  ER-negative breast cancer cases European</span></code></pre></div>
<p>Again, not so surprisingly, all samples are of European ancestry, a bias recognized by the research community. You can find more details about the ancestry variable in the <code>ancestry_categories</code> column. These categories have been defined within the NHGRI-EBI GWAS Catalog framework. We provide these ancestry nomenclature in <em>quincunx</em> as a separate dataset named <code>ancestry_categories</code>. See <code><a href="../reference/ancestry_categories.html">?ancestry_categories</a></code> more details. To quickly lookup the definition of the <em>European</em> ancestry:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Quick look at the ancestries definitions</span>
<span class="va">ancestry_categories</span>
<span class="co">#&gt; # A tibble: 17 x 3</span>
<span class="co">#&gt;    ancestry                    definition                       examples        </span>
<span class="co">#&gt;    &lt;chr&gt;                       &lt;chr&gt;                            &lt;chr&gt;           </span>
<span class="co">#&gt;  1 Aboriginal Australian       "Includes individuals who eithe… Martu Australia…</span>
<span class="co">#&gt;  2 African American or Afro-C… "Includes individuals who eithe… African America…</span>
<span class="co">#&gt;  3 African unspecified         "Includes individuals that eith… African, non-Hi…</span>
<span class="co">#&gt;  4 Asian unspecified           "Includes individuals that eith… Asian, Asian Am…</span>
<span class="co">#&gt;  5 Central Asian               "Includes individuals who eithe… Silk Road (foun…</span>
<span class="co">#&gt;  6 East Asian                  "Includes individuals who eithe… Chinese, Japane…</span>
<span class="co">#&gt;  7 European                    "Includes individuals who eithe… Spanish, Swedish</span>
<span class="co">#&gt;  8 Greater Middle Eastern (Mi… "Includes individuals who self-… Tunisian, Arab,…</span>
<span class="co">#&gt;  9 Hispanic or Latin American  "Includes individuals who eithe… Brazilian, Mexi…</span>
<span class="co">#&gt; 10 Native American             "Includes indigenous individual… Pima Indian, Pl…</span>
<span class="co">#&gt; 11 Not reported                "Includes individuals for which… &lt;NA&gt;            </span>
<span class="co">#&gt; 12 Oceanian                    "Includes individuals that eith… Solomon Islande…</span>
<span class="co">#&gt; 13 Other                       "Includes individuals where an … Surinamese, Rus…</span>
<span class="co">#&gt; 14 Other admixed ancestry      "Includes individuals who eithe… &lt;NA&gt;            </span>
<span class="co">#&gt; 15 South Asian                 "Includes individuals who eithe… Bangladeshi, Sr…</span>
<span class="co">#&gt; 16 South East Asian            "Includes individuals who eithe… Thai, Malay     </span>
<span class="co">#&gt; 17 Sub-Saharan African         "Includes individuals who eithe… Yoruban, Gambian</span></code></pre></div>
</div>
<div id="scores-object-demographics-table" class="section level3">
<h3 class="hasAnchor">
<a href="#scores-object-demographics-table" class="anchor"></a>scores object | demographics table</h3>
<p>The demographics table usually lists demographic information about each sample. For this study this table is however empty, meaning that this information was either not available from Mavaddat’s publication, or not included in the PGS Catalog.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">demographics</span>
<span class="co">#&gt; # A tibble: 0 x 11</span>
<span class="co">#&gt; # … with 11 variables: pgs_id &lt;chr&gt;, sample_id &lt;int&gt;, variable &lt;chr&gt;,</span>
<span class="co">#&gt; #   estimate_type &lt;chr&gt;, estimate &lt;dbl&gt;, unit &lt;chr&gt;, variability_type &lt;chr&gt;,</span>
<span class="co">#&gt; #   variability &lt;dbl&gt;, interval_type &lt;chr&gt;, interval_lower &lt;dbl&gt;,</span>
<span class="co">#&gt; #   interval_upper &lt;dbl&gt;</span></code></pre></div>
<p>Nevertheless, the demographics variables, when present, are follow-up time, and age of study participants.</p>
<p>If you want to confirm that <em>quincunx</em> is retrieving exactly the same info as provided by the PGS Catalog web interface, you can always check this by showing online the metadata for your PGP publication of interest using the function <code>open_in_pgs_catalog</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/open_in_pgs_catalog.html">open_in_pgs_catalog</a></span><span class="op">(</span><span class="st">'PGP000002'</span>, pgs_catalog_entity <span class="op">=</span> <span class="st">'pgp'</span><span class="op">)</span></code></pre></div>
</div>
<div id="scores-object-cohorts-table" class="section level3">
<h3 class="hasAnchor">
<a href="#scores-object-cohorts-table" class="anchor"></a>scores object | cohorts table</h3>
<p>In the cohorts table you can check which cohorts are associated with which samples. Note that the unique identification of a sample is given by the combination of the values of the first two columns: <code>pgs_id</code>, and <code>sample_id</code>.<br>
To learn more about the meaning of cohorts for the PGS Catalog, visit our check our <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf">cheatsheet</a>, section <em>Cohorts, Samples and Sample Sets</em>, second page.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">cohorts</span>
<span class="co">#&gt; # A tibble: 558 x 4</span>
<span class="co">#&gt;    pgs_id    sample_id cohort_symbol cohort_name                                </span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;         &lt;chr&gt;                                      </span>
<span class="co">#&gt;  1 PGS000004         2 ABCFS         Australian Breast Cancer Family Study      </span>
<span class="co">#&gt;  2 PGS000004         2 MCCS          Melbourne Collaborative Cohort Study       </span>
<span class="co">#&gt;  3 PGS000004         2 HMBCS         Hannover-Minsk Breast Cancer Study         </span>
<span class="co">#&gt;  4 PGS000004         2 LMBC          Leuven Multidisciplinary Breast Centre     </span>
<span class="co">#&gt;  5 PGS000004         2 MTLGEBCS      Montreal Gene-Environment Breast Cancer St…</span>
<span class="co">#&gt;  6 PGS000004         2 CGPS          Copenhagen General Population Study        </span>
<span class="co">#&gt;  7 PGS000004         2 KBCP          Kuopio Breast Cancer Project               </span>
<span class="co">#&gt;  8 PGS000004         2 OBCS          Oulu Breast Cancer Study                   </span>
<span class="co">#&gt;  9 PGS000004         2 CECILE        CECILE Breast Cancer Study                 </span>
<span class="co">#&gt; 10 PGS000004         2 BBCC          Bavarian Breast Cancer Cases and Controls  </span>
<span class="co">#&gt; # … with 548 more rows</span></code></pre></div>
</div>
<div id="scores-object-traits-table" class="section level3">
<h3 class="hasAnchor">
<a href="#scores-object-traits-table" class="anchor"></a>scores object | traits table</h3>
<p>Finally, in the traits table, you have access to the traits (phenotypes) associated with these polygenic scores. In this study, all scores indicate “breast cancer” or one of its subtypes (column <code>trait</code>):</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">traits</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'efo_id'</span>, <span class="st">'trait'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 6 x 3</span>
<span class="co">#&gt;   pgs_id    efo_id      trait                                   </span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;       &lt;chr&gt;                                   </span>
<span class="co">#&gt; 1 PGS000004 EFO_0000305 breast carcinoma                        </span>
<span class="co">#&gt; 2 PGS000005 EFO_1000649 estrogen-receptor positive breast cancer</span>
<span class="co">#&gt; 3 PGS000006 EFO_1000650 estrogen-receptor negative breast cancer</span>
<span class="co">#&gt; 4 PGS000007 EFO_0000305 breast carcinoma                        </span>
<span class="co">#&gt; 5 PGS000008 EFO_1000649 estrogen-receptor positive breast cancer</span>
<span class="co">#&gt; 6 PGS000009 EFO_1000650 estrogen-receptor negative breast cancer</span></code></pre></div>
<p>Compared to the author-reported trait (column <code>reported_trait</code> from table <code>scores</code>), the trait description in this table follows the controlled vocabulary of an ontology, i.e., the <a href="https://www.ebi.ac.uk/efo/">Experimental Factor Ontology (EFO)</a>. This way, traits are described objectively. This is very useful for comparing trait data among different studies where different reported trait descriptions might have been used. For example, if you want now to know what other polygenic scores may be deposited in the PGS Catalog that also study breast cancer — namely, breast carcinoma, estrogen-receptor positive breast cancer, or estrogen-receptor negative breast cancer — then you could use their respective EFO identifiers (EFO_0000305, EFO_1000649, or EFO_1000650) with the function <code><a href="../reference/get_scores.html">get_scores()</a></code>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scores_bc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_scores.html">get_scores</a></span><span class="op">(</span>efo_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">traits</span><span class="op">[[</span><span class="st">'efo_id'</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">scores_bc</span><span class="op">)</span>
<span class="co">#&gt; [1] 96</span></code></pre></div>
<p>So there are 96 scores present in the PGS Catalog. Included in this set are the 6 scores originating from <a href="https://doi.org/10.1016/j.ajhg.2018.11.002">Mavaddat et al. (2018)</a>. If we want to proceed with analysing these other scores without including those from the Mavaddat’s study, we could use the function <code><a href="../reference/setop.html">setdiff()</a></code> to remove them from the S4 <code>scores</code> object:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use quincunx::setdiff to avoid collision with dplyr::setdiff()</span>
<span class="va">bc_scores_not_mavaddat2018</span> <span class="op">&lt;-</span> <span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/setop.html">setdiff</a></span><span class="op">(</span><span class="va">scores_bc</span>, <span class="va">mavaddat2018_scores</span><span class="op">)</span>
<span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">bc_scores_not_mavaddat2018</span><span class="op">)</span>
<span class="co">#&gt; [1] 90</span>
<span class="va">bc_scores_not_mavaddat2018</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'reported_trait'</span>,<span class="st">'n_variants'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 90 x 3</span>
<span class="co">#&gt;    pgs_id    reported_trait n_variants</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;               &lt;int&gt;</span>
<span class="co">#&gt;  1 PGS000001 Breast Cancer          77</span>
<span class="co">#&gt;  2 PGS000028 Breast cancer          83</span>
<span class="co">#&gt;  3 PGS000029 Breast cancer          76</span>
<span class="co">#&gt;  4 PGS000045 Breast cancer          88</span>
<span class="co">#&gt;  5 PGS000050 Breast cancer          44</span>
<span class="co">#&gt;  6 PGS000052 Breast cancer         161</span>
<span class="co">#&gt;  7 PGS000072 Breast cancer         187</span>
<span class="co">#&gt;  8 PGS000153 Breast cancer          66</span>
<span class="co">#&gt;  9 PGS000015 Breast cancer        5218</span>
<span class="co">#&gt; 10 PGS000317 Breast cancer         180</span>
<span class="co">#&gt; # … with 80 more rows</span></code></pre></div>
<p>For other set operations, check their documentation page: <code><a href="../reference/setop.html">union()</a></code>, <code><a href="../reference/setop.html">intersect()</a></code> and <code><a href="../reference/setop.html">setequal()</a></code>.</p>
</div>
</div>
<div id="performance-metrics" class="section level2">
<h2 class="hasAnchor">
<a href="#performance-metrics" class="anchor"></a>Performance Metrics</h2>
<p>According to the PGS Catalog team (<a href="https://doi.org/10.1101/2020.05.20.20108217" class="uri">https://doi.org/10.1101/2020.05.20.20108217</a>):</p>
<blockquote>
<p>“Performance Metrics assess the validity of a PGS in a Sample Set, independent of the samples used for score development. Common metrics include standardised effect sizes (odds/hazard ratios [OR/HR], and regression coefficients [Beta]), classification accuracy metrics (e.g. AUROC, C-index, AUPRC), but other relevant metrics (e.g. calibration [Chi-square]) can also be recorded. The covariates used in the model (most commonly age, sex, and genetic principal components (PCs) to account of population structure) are also linking to each set of metrics. Multiple PGS can be evaluated on the same Sample Set and further indexed as directly comparable Performance Metrics.”</p>
</blockquote>
<p>In this section we will learn how to retrieve details about the evaluation of the polygenic scores developed in Mavaddat <em>et al.</em> (2018). To do this, we start by querying the performance metrics for our scores of interest.</p>
<p>Please recall that the <code>pgs_ids</code> and <code>reported_traits</code> for the polygenic scores newly developed in Mavaddat’s publication are:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'reported_trait'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   pgs_id    reported_trait           </span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;                    </span>
<span class="co">#&gt; 1 PGS000004 Breast Cancer            </span>
<span class="co">#&gt; 2 PGS000005 ER-positive Breast Cancer</span>
<span class="co">#&gt; 3 PGS000006 ER-negative Breast Cancer</span>
<span class="co">#&gt; 4 PGS000007 Breast Cancer            </span>
<span class="co">#&gt; 5 PGS000008 ER-positive Breast Cancer</span>
<span class="co">#&gt; 6 PGS000009 ER-negative Breast Cancer</span></code></pre></div>
<p>So we use these PGS identifiers to query the PGS Catalog for performance metrics using the function <code><a href="../reference/get_performance_metrics.html">get_performance_metrics()</a></code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_performance_metrics.html">get_performance_metrics</a></span><span class="op">(</span>pgs_id <span class="op">=</span> <span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">$</span><span class="va">pgs_id</span><span class="op">)</span></code></pre></div>
<p>The output is an S4 object with 9 tables:</p>
<ul>
<li>performance_metrics, publications, sample_sets, samples, demographics, cohorts, pgs_effect_sizes, pgs_classification_metrics, and pgs_other_metrics.</li>
</ul>
<p>Reminder | You can quickly check all variables from each table by consulting <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf">quincunx cheatsheet</a>.<br>
Also, a description of each variable is annotated in the <code>performance_metrics</code> object help page that can be accessed with <code><a href="../reference/performance_metrics-class.html">class?performance_metrics</a></code>.</p>
<div id="performance_metrics-object-performance-metrics-table" class="section level3">
<h3 class="hasAnchor">
<a href="#performance_metrics-object-performance-metrics-table" class="anchor"></a>performance_metrics object | Performance Metrics table</h3>
<p>In the first table <code>performance_metrics</code> we get one performance metrics entity per row, with the following columns: ppm_id, pgs_id, reported_trait, covariates, comments.<br>
Note that all performance metrics are indexed with an unique identifier that starts with “PPM”.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">performance_metrics</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="co">#&gt; # A tibble: 74 x 4</span>
<span class="co">#&gt;    ppm_id  pgs_id  reported_trait                 covariates                    </span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;                          &lt;chr&gt;                         </span>
<span class="co">#&gt;  1 PPM000… PGS000… Invasive breast cancer         study, genetic PCs 1-15       </span>
<span class="co">#&gt;  2 PPM000… PGS000… Incident breast cancer cases   study, genetic PCs 1-15       </span>
<span class="co">#&gt;  3 PPM000… PGS000… Breast cancer intrinsic-like … &lt;NA&gt;                          </span>
<span class="co">#&gt;  4 PPM000… PGS000… Breast cancer intrinsic-like … &lt;NA&gt;                          </span>
<span class="co">#&gt;  5 PPM000… PGS000… Breast cancer intrinsic-like … &lt;NA&gt;                          </span>
<span class="co">#&gt;  6 PPM000… PGS000… Breast cancer intrinsic-like … &lt;NA&gt;                          </span>
<span class="co">#&gt;  7 PPM000… PGS000… Breast cancer intrinsic-like … &lt;NA&gt;                          </span>
<span class="co">#&gt;  8 PPM000… PGS000… Metachronous contralateral br… Country                       </span>
<span class="co">#&gt;  9 PPM000… PGS000… Metachronous contralateral br… Country, age at first diagnos…</span>
<span class="co">#&gt; 10 PPM000… PGS000… Invasive metachronous contral… Country                       </span>
<span class="co">#&gt; # … with 64 more rows</span></code></pre></div>
<p>According to the PGS Catalog documentation (<a href="http://www.pgscatalog.org/docs/" class="uri">http://www.pgscatalog.org/docs/</a>):</p>
<ul>
<li>The <code>reported_trait</code> displays the reported trait, often corresponding to the test set names reported in the publication, or more specific aspects of the phenotype being tested (e.g. if the disease cases are incident vs. recurrent events).<br>
</li>
<li>The <code>covariates</code> column lists the covariates used in the prediction model to evaluate the PGS. Examples include: age, sex, smoking habits, etc.<br>
</li>
<li>The <code>comments</code> column is a field where additional relevant information can be added to aid with understanding a particular performance metrics.</li>
</ul>
<p>Looking at the <code>performance_metrics</code> table, we can see that one polygenic score (<code>pgs_id</code>) can be associated with several performance metrics (<code>ppm_id</code>), e.g., PGS000007 associates with 4 PPMs:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">performance_metrics</span>, <span class="va">pgs_id</span> <span class="op">==</span> <span class="st">'PGS000007'</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 4 x 5</span>
<span class="co">#&gt;   ppm_id    pgs_id    reported_trait               covariates           comments</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;                        &lt;chr&gt;                &lt;chr&gt;   </span>
<span class="co">#&gt; 1 PPM000008 PGS000007 Invasive breast cancer       study, genetic PCs … &lt;NA&gt;    </span>
<span class="co">#&gt; 2 PPM000384 PGS000007 Breast Cancer (personal his… age at menarche      &lt;NA&gt;    </span>
<span class="co">#&gt; 3 PPM000386 PGS000007 Breast Cancer (personal his… age, sex             &lt;NA&gt;    </span>
<span class="co">#&gt; 4 PPM000388 PGS000007 Breast Cancer (personal his… &lt;NA&gt;                 &lt;NA&gt;</span></code></pre></div>
<p>This means that the polygenic score (PGS000007) has been validated several times, using different models. In this case, we can immediately see that PGS000007 performance has been evaluated, for example, for alternative breast cancer types (different <code>reported_traits</code>), namely:</p>
<ul>
<li>Invasive breast cancer, Breast Cancer (personal history).</li>
</ul>
<p>Additionally, we can see that the same <code>reported_trait</code> (<em>Breast Cancer (personal history)</em>) has been validated by 3 different performance metrics (PPMs): PPM000384, PPM000386, PPM000388; each having included different covariates in the model: age at menarche, age, sex, NA. (NA means that data for this field is <em>Not Available</em> in the records).</p>
</div>
<div id="performance_metrics-object-publications-table" class="section level3">
<h3 class="hasAnchor">
<a href="#performance_metrics-object-publications-table" class="anchor"></a>performance_metrics object | Publications table</h3>
<p>The publications table is dedicated to hold information related to the publications associated with the performance metrics. The column <code>pgp_id</code> links each performance metrics to the respective publication where that performance metrics was reported and collected.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">publications</span>
<span class="co">#&gt; # A tibble: 74 x 8</span>
<span class="co">#&gt;    ppm_id pgp_id pubmed_id publication_date publication title author_fullname</span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;     &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;          </span>
<span class="co">#&gt;  1 PPM00… PGP00… 30554720  2018-12-13       Am J Hum G… Poly… Mavaddat N     </span>
<span class="co">#&gt;  2 PPM00… PGP00… 30554720  2018-12-13       Am J Hum G… Poly… Mavaddat N     </span>
<span class="co">#&gt;  3 PPM00… PGP00… 32424353  2020-05-18       Nat Genet   Geno… Zhang H        </span>
<span class="co">#&gt;  4 PPM00… PGP00… 32424353  2020-05-18       Nat Genet   Geno… Zhang H        </span>
<span class="co">#&gt;  5 PPM00… PGP00… 32424353  2020-05-18       Nat Genet   Geno… Zhang H        </span>
<span class="co">#&gt;  6 PPM00… PGP00… 32424353  2020-05-18       Nat Genet   Geno… Zhang H        </span>
<span class="co">#&gt;  7 PPM00… PGP00… 32424353  2020-05-18       Nat Genet   Geno… Zhang H        </span>
<span class="co">#&gt;  8 PPM00… PGP00… 33022221  2020-10-05       Am J Hum G… Brea… Kramer I       </span>
<span class="co">#&gt;  9 PPM00… PGP00… 33022221  2020-10-05       Am J Hum G… Brea… Kramer I       </span>
<span class="co">#&gt; 10 PPM00… PGP00… 33022221  2020-10-05       Am J Hum G… Brea… Kramer I       </span>
<span class="co">#&gt; # … with 64 more rows, and 1 more variable: doi &lt;chr&gt;</span></code></pre></div>
<p>Here we can immediately see that there are more publications than just the Mavaddat <em>et al.</em> (pubmed_id = 30554720) that we started with.<br>
This is expected because we requested all the performance metrics for the PGS scores that were newly <strong>developed</strong> by Mavaddat <em>et al.</em>; but these scores have been subsequently <strong>evaluated</strong> by other posterior publications, and accordingly have performance metrics reported in these posterior evaluations.<br>
This is easily confirmed by checking that all other publications are dated after December 13th 2018, which is the date of publication of the original Mavaddat <em>et al.</em> paper.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">publication_date</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; [1] "2018-12-13" "2020-05-18" "2020-10-05" "2020-07-15" "2020-12-14"</span>
<span class="co">#&gt; [6] "2020-03-12" "2019-11-26"</span></code></pre></div>
<p>We can choose to look at those publications later to see what evaluations (performance metrics) were reported in them (for which traits, adjusting for which covariates, etc.).</p>
<p>But for now, we are only interested in studying the performance metrics reported by Mavaddat <em>et al.</em> for the PGS scores newly developed in this publication. So, lets proceed with creating a vector containing only the <code>ppm_ids</code> of interest. We will do this by filtering the <code>ppm_ids</code> for the <code>pubmed_id</code> corresponding to the Mavaddat publication. We will then use this vector to subset the following tables (to display only the metrics for these PPMs of interest).</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm_ids</span> <span class="op">&lt;-</span> <span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">publications</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">pubmed_id</span> <span class="op">==</span> <span class="st">"30554720"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">ppm_id</span><span class="op">)</span>
<span class="va">mavaddat2018_ppm_ids</span>
<span class="co">#&gt; [1] "PPM000004" "PPM000005" "PPM000006" "PPM000007" "PPM000008" "PPM000009"</span>
<span class="co">#&gt; [7] "PPM000010"</span>

<span class="co"># Find the corresponding pgs_id, reported_trait, covariates, and comments</span>
<span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">performance_metrics</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 7 x 5</span>
<span class="co">#&gt;   ppm_id   pgs_id   reported_trait         covariates       comments            </span>
<span class="co">#&gt;   &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;                  &lt;chr&gt;            &lt;chr&gt;               </span>
<span class="co">#&gt; 1 PPM0000… PGS0000… Invasive breast cancer study, genetic … &lt;NA&gt;                </span>
<span class="co">#&gt; 2 PPM0000… PGS0000… Incident breast cance… study, genetic … Included only 306 o…</span>
<span class="co">#&gt; 3 PPM0000… PGS0000… ER-positive breast ca… study, genetic … &lt;NA&gt;                </span>
<span class="co">#&gt; 4 PPM0000… PGS0000… ER-negative breast ca… study, genetic … &lt;NA&gt;                </span>
<span class="co">#&gt; 5 PPM0000… PGS0000… Invasive breast cancer study, genetic … &lt;NA&gt;                </span>
<span class="co">#&gt; 6 PPM0000… PGS0000… ER-positive breast ca… study, genetic … &lt;NA&gt;                </span>
<span class="co">#&gt; 7 PPM0000… PGS0000… ER-negative breast ca… study, genetic … &lt;NA&gt;</span></code></pre></div>
<p>Here, we can immediately see that the Mavaddat publication has evaluated the PGS000004 twice, with two performance metrics (PPM000004 and PPM000005), that are different because they include a different set of SNPs (see the <code>comments</code> column for PPM000005).</p>
</div>
<div id="performance_metrics-object-sample-sets-table" class="section level3">
<h3 class="hasAnchor">
<a href="#performance_metrics-object-sample-sets-table" class="anchor"></a>performance_metrics object | Sample Sets table</h3>
<p>The PGS Catalog provides a Sample Set Id (PSS) that links the PPMs to the sample sets that were used to evaluate the corresponding PGS. This mapping is stored in the <code>sample_sets</code> table.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">sample_sets</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span> 
<span class="co">#&gt; # A tibble: 7 x 2</span>
<span class="co">#&gt;   ppm_id    pss_id   </span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;    </span>
<span class="co">#&gt; 1 PPM000004 PSS000004</span>
<span class="co">#&gt; 2 PPM000005 PSS000007</span>
<span class="co">#&gt; 3 PPM000006 PSS000005</span>
<span class="co">#&gt; 4 PPM000007 PSS000006</span>
<span class="co">#&gt; 5 PPM000008 PSS000004</span>
<span class="co">#&gt; 6 PPM000009 PSS000005</span>
<span class="co">#&gt; 7 PPM000010 PSS000006</span></code></pre></div>
</div>
<div id="performance_metrics-object-samples-table" class="section level3">
<h3 class="hasAnchor">
<a href="#performance_metrics-object-samples-table" class="anchor"></a>performance_metrics object | Samples table</h3>
<p>The <code>samples</code> table gathers more relevant information regarding the samples used for the relevant evaluations. This table contains the following columns:</p>
<ul>
<li>ppm_id, pss_id, sample_id, stage, sample_size, sample_cases, sample_controls, sample_percent_male, phenotype_description, ancestry, ancestry_description, ancestry_country, ancestry_additional_description, study_id, pubmed_id, cohorts_additional_description.</li>
</ul>
<p>Please note that the samples are not identified in PGS Catalog with a global unique identifier, but <em>quincunx</em> assigns a surrogate identifier (<code>sample_id</code>) to allow the mapping between tables.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">samples</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 7 x 16</span>
<span class="co">#&gt;   ppm_id pss_id sample_id stage sample_size sample_cases sample_controls</span>
<span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;      &lt;int&gt; &lt;chr&gt;       &lt;int&gt;        &lt;int&gt;           &lt;int&gt;</span>
<span class="co">#&gt; 1 PPM00… PSS00…         1 eval…       29751        11428           18323</span>
<span class="co">#&gt; 2 PPM00… PSS00…         1 eval…      190040         3215          186825</span>
<span class="co">#&gt; 3 PPM00… PSS00…         1 eval…       11428         7992            3436</span>
<span class="co">#&gt; 4 PPM00… PSS00…         1 eval…       11428         1259           10169</span>
<span class="co">#&gt; 5 PPM00… PSS00…         1 eval…       29751        11428           18323</span>
<span class="co">#&gt; 6 PPM00… PSS00…         1 eval…       11428         7992            3436</span>
<span class="co">#&gt; 7 PPM00… PSS00…         1 eval…       11428         1259           10169</span>
<span class="co">#&gt; # … with 9 more variables: sample_percent_male &lt;dbl&gt;,</span>
<span class="co">#&gt; #   phenotype_description &lt;chr&gt;, ancestry &lt;chr&gt;, ancestry_description &lt;chr&gt;,</span>
<span class="co">#&gt; #   ancestry_country &lt;chr&gt;, ancestry_additional_description &lt;chr&gt;,</span>
<span class="co">#&gt; #   study_id &lt;chr&gt;, pubmed_id &lt;chr&gt;, cohorts_additional_description &lt;chr&gt;</span></code></pre></div>
<p>Here we can, for example see that the sample sizes are very different for each evaluation. Let’s take a quick look at their values.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">samples</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">ppm_id</span>, <span class="va">sample_cases</span>, <span class="va">sample_controls</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span><span class="op">!</span><span class="va">ppm_id</span>, names_to <span class="op">=</span> <span class="st">"sample_type"</span>, values_to <span class="op">=</span> <span class="st">"count"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">sample_type</span>, y<span class="op">=</span><span class="va">count</span>, x<span class="op">=</span><span class="va">ppm_id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">20</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="pgs-scores-mavaddat_files/figure-html/ppm_samples_plot-1.png" width="672" style="display: block; margin: auto;"></p>
<p>This plot clearly shows that PPM000005 uses a very large sample (particularly regarding the number of controls) when compared with all other reported PPMs.</p>
</div>
<div id="performance_metrics-object-demographics-table" class="section level3">
<h3 class="hasAnchor">
<a href="#performance_metrics-object-demographics-table" class="anchor"></a>performance_metrics object | Demographics table</h3>
<p>The <code>demographics</code> table holds information regarding the demographics’ variables of each <code>sample</code>. Each demographics’ variable (row) is uniquely identified by the combination of values from the columns: <code>ppm_id</code>, <code>pss_id</code>, <code>sample_id</code>, and <code>variable</code>. Currently, the PGS Catalog only describes two demographic variables: <em>age of participants</em> and <em>follow-up time</em>.</p>
<p>The columns presented in the table are:</p>
<ul>
<li>ppm_id, pss_id, sample_id, variable, estimate_type, estimate, unit, variability_type, variability, interval_type, interval_lower, interval_upper</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">demographics</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Rows: 2</span>
<span class="co">#&gt; Columns: 12</span>
<span class="co">#&gt; $ ppm_id           &lt;chr&gt; "PPM001345", "PPM001347"</span>
<span class="co">#&gt; $ pss_id           &lt;chr&gt; "PSS000450", "PSS000450"</span>
<span class="co">#&gt; $ sample_id        &lt;int&gt; 1, 1</span>
<span class="co">#&gt; $ variable         &lt;chr&gt; "age", "age"</span>
<span class="co">#&gt; $ estimate_type    &lt;chr&gt; "mean age (at the end of follow-up)", "mean age (at …</span>
<span class="co">#&gt; $ estimate         &lt;dbl&gt; 58.5, 58.5</span>
<span class="co">#&gt; $ unit             &lt;chr&gt; "years", "years"</span>
<span class="co">#&gt; $ variability_type &lt;chr&gt; NA, NA</span>
<span class="co">#&gt; $ variability      &lt;dbl&gt; NA, NA</span>
<span class="co">#&gt; $ interval_type    &lt;chr&gt; "iqr", "iqr"</span>
<span class="co">#&gt; $ interval_lower   &lt;dbl&gt; 45.1, 45.1</span>
<span class="co">#&gt; $ interval_upper   &lt;dbl&gt; 72.2, 72.2</span></code></pre></div>
<p>Here we can see that neither of the PPMs shown is from the Mavaddat paper. This means that the performance metrics reported in the paper do not have any information regarding the demographics’ variables that belong in this table (age and follow-up time).</p>
</div>
<div id="performance_metrics-object-cohorts-table" class="section level3">
<h3 class="hasAnchor">
<a href="#performance_metrics-object-cohorts-table" class="anchor"></a>performance_metrics object | Cohorts table</h3>
<p>Similarly to the <code>cohorts</code> table described above (in the <code>scores</code> object class), this table shows which cohorts are associated with each sample. However here, the unique identification of a sample can be obtained by combining the values of the first three columns: <code>ppm_id</code>, <code>pss_id</code> and <code>sample_id</code>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">cohorts</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 61 x 5</span>
<span class="co">#&gt;    ppm_id   pss_id  sample_id cohort_symbol cohort_name                         </span>
<span class="co">#&gt;    &lt;chr&gt;    &lt;chr&gt;       &lt;int&gt; &lt;chr&gt;         &lt;chr&gt;                               </span>
<span class="co">#&gt;  1 PPM0000… PSS000…         1 AHS           Agricultural Health Study           </span>
<span class="co">#&gt;  2 PPM0000… PSS000…         1 BGS           Breakthrough Generations Study      </span>
<span class="co">#&gt;  3 PPM0000… PSS000…         1 EPIC          European Prospective Investigation …</span>
<span class="co">#&gt;  4 PPM0000… PSS000…         1 FHRISK        Family History Clinic in Manchester </span>
<span class="co">#&gt;  5 PPM0000… PSS000…         1 KARMA         UNKNOWN                             </span>
<span class="co">#&gt;  6 PPM0000… PSS000…         1 NHS           Nurses Health Study                 </span>
<span class="co">#&gt;  7 PPM0000… PSS000…         1 NHS2          Nurses Health Studies II            </span>
<span class="co">#&gt;  8 PPM0000… PSS000…         1 PLCO          Prostate, Lung, Colorectal, and Ova…</span>
<span class="co">#&gt;  9 PPM0000… PSS000…         1 PROCAS        Breast Screening Programme (NHSBSP)…</span>
<span class="co">#&gt; 10 PPM0000… PSS000…         1 SISTER        UNKNOWN                             </span>
<span class="co">#&gt; # … with 51 more rows</span></code></pre></div>
</div>
<div id="performance_metrics-object-pgs-effect-sizes-pgs-classification-metrics-pgs-other-metrics-tables" class="section level3">
<h3 class="hasAnchor">
<a href="#performance_metrics-object-pgs-effect-sizes-pgs-classification-metrics-pgs-other-metrics-tables" class="anchor"></a>performance_metrics object | PGS Effect Sizes &amp; PGS Classification Metrics &amp; PGS Other Metrics tables</h3>
<p>The three final tables of the <code>performance_metrics</code> object hold the performance metrics themselves used in the validation. Each table presents the same column structure with 11 total columns, where the second column is different between the three tables. This column shows an id created by <em>quincunx</em> that is used to identify each of the individual metrics tables (<code>effect_size_id</code>, <code>classification_metrics_id</code>, or <code>other_metrics_id</code>). All other columns are for the same variables:</p>
<ul>
<li>ppm_id, effect_size_id, estimate_type_long, estimate_type, estimate, unit, variability_type, variability, interval_type, interval_lower, interval_upper.</li>
</ul>
<p>(These column names are specifically from the <code>pgs_effect_sizes</code> table, as indicated by the second column name. All other columns are equally named in all three tables).</p>
<p>According to the PGS Catalog online documentation (<a href="http://www.pgscatalog.org/docs/" class="uri">http://www.pgscatalog.org/docs/</a>):</p>
<blockquote>
<p>“The reported values of the performance metrics are all reported similarly (e.g. the estimate is recorded along with the 95% confidence interval (if supplied)) and grouped according to the type of statistic they represent:<br>
- <strong>PGS Effect Sizes (per SD change)</strong> | Standardized effect sizes, per standard deviation [SD] change in PGS. Examples include regression coefficients (Betas) for continuous traits, Odds ratios (OR) and/or Hazard ratios (HR) for dichotomous traits depending on the availability of time-to-event data.<br>
- <strong>PGS Classification Metrics</strong> | Examples include the Area under the Receiver Operating Characteristic (AUROC) or Harrell’s C-index (Concordance statistic).<br>
- <strong>Other Metrics</strong> | Metrics that do not fit into the other two categories. Examples include: R2 (proportion of the variance explained), or reclassification metrics.”</p>
</blockquote>
<p>Now, lets briefly explore the data in these tables (as usual filtered for only the PPMs that were newly reported in Mavaddat <em>et al.</em>).</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">pgs_effect_sizes</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 7 x 11</span>
<span class="co">#&gt;   ppm_id effect_size_id estimate_type_l… estimate_type estimate unit </span>
<span class="co">#&gt;   &lt;chr&gt;           &lt;int&gt; &lt;chr&gt;            &lt;chr&gt;            &lt;dbl&gt; &lt;chr&gt;</span>
<span class="co">#&gt; 1 PPM00…              1 Odds Ratio       OR                1.61 &lt;NA&gt; </span>
<span class="co">#&gt; 2 PPM00…              1 Hazard Ratio     HR                1.59 &lt;NA&gt; </span>
<span class="co">#&gt; 3 PPM00…              1 Odds Ratio       OR                1.68 &lt;NA&gt; </span>
<span class="co">#&gt; 4 PPM00…              1 Odds Ratio       OR                1.45 &lt;NA&gt; </span>
<span class="co">#&gt; 5 PPM00…              1 Odds Ratio       OR                1.66 &lt;NA&gt; </span>
<span class="co">#&gt; 6 PPM00…              1 Odds Ratio       OR                1.73 &lt;NA&gt; </span>
<span class="co">#&gt; 7 PPM00…              1 Odds Ratio       OR                1.44 &lt;NA&gt; </span>
<span class="co">#&gt; # … with 5 more variables: variability_type &lt;chr&gt;, variability &lt;dbl&gt;,</span>
<span class="co">#&gt; #   interval_type &lt;chr&gt;, interval_lower &lt;dbl&gt;, interval_upper &lt;dbl&gt;</span>
<span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">pgs_classification_metrics</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 11</span>
<span class="co">#&gt;   ppm_id classification_… estimate_type_l… estimate_type estimate unit </span>
<span class="co">#&gt;   &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;            &lt;chr&gt;            &lt;dbl&gt; &lt;chr&gt;</span>
<span class="co">#&gt; 1 PPM00…                1 Area Under the … AUROC            0.63  &lt;NA&gt; </span>
<span class="co">#&gt; 2 PPM00…                1 Area Under the … AUROC            0.641 &lt;NA&gt; </span>
<span class="co">#&gt; 3 PPM00…                1 Area Under the … AUROC            0.601 &lt;NA&gt; </span>
<span class="co">#&gt; 4 PPM00…                1 Area Under the … AUROC            0.636 &lt;NA&gt; </span>
<span class="co">#&gt; 5 PPM00…                1 Area Under the … AUROC            0.647 &lt;NA&gt; </span>
<span class="co">#&gt; 6 PPM00…                1 Area Under the … AUROC            0.6   &lt;NA&gt; </span>
<span class="co">#&gt; # … with 5 more variables: variability_type &lt;chr&gt;, variability &lt;dbl&gt;,</span>
<span class="co">#&gt; #   interval_type &lt;chr&gt;, interval_lower &lt;dbl&gt;, interval_upper &lt;dbl&gt;</span>
<span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">pgs_other_metrics</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op">%in%</span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 0 x 11</span>
<span class="co">#&gt; # … with 11 variables: ppm_id &lt;chr&gt;, other_metrics_id &lt;int&gt;,</span>
<span class="co">#&gt; #   estimate_type_long &lt;chr&gt;, estimate_type &lt;chr&gt;, estimate &lt;dbl&gt;, unit &lt;chr&gt;,</span>
<span class="co">#&gt; #   variability_type &lt;chr&gt;, variability &lt;dbl&gt;, interval_type &lt;chr&gt;,</span>
<span class="co">#&gt; #   interval_lower &lt;dbl&gt;, interval_upper &lt;dbl&gt;</span></code></pre></div>
<p>We can immediately see that the third table (reserved for metrics other than effect sizes and classification metrics) is empty; and that the effect sizes were estimated using Odds Ratio and Hazard Ratio, and the classification metric applied was AUROC.</p>
<p>Now we can look at the reported values for each PPM metrics and decide if the validation is relevant, and therefore make an informed choice to use the associated PGS score for our own study, and eventually apply it to our own dataset.</p>
</div>
<div id="concluding-remarks" class="section level3">
<h3 class="hasAnchor">
<a href="#concluding-remarks" class="anchor"></a>Concluding remarks</h3>
<p>Useful reminders:</p>
<ul>
<li>You can quickly check all variables from each table by consulting <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf">quincunx cheatsheet</a>.<br>
</li>
<li>All <em>quincunx</em> functions are annotated with help pages accessed with <code>?function_name</code> (e.g. <code><a href="../reference/get_scores.html">?get_scores</a></code>, <code><a href="../reference/get_traits.html">?get_traits</a></code>, <code><a href="../reference/get_performance_metrics.html">?get_performance_metrics</a></code>).<br>
</li>
<li>All tables retrieved by <em>quincunx’s</em> functions are annotated with help files that describe all variables present. These are accessible via <code>class?object_name</code> (e.g. <code><a href="../reference/scores-class.html">class?scores</a></code>, <code><a href="../reference/performance_metrics-class.html">class?performance_metrics</a></code>).</li>
<li>The <em>quincunx</em> package has been published in a peer-reviewed journal. Use <code><a href="https://rdrr.io/r/utils/citation.html">citation(package="quincunx")</a></code> to get the full paper citation.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://rmagno.eu/">Ramiro Magno</a>, <a href="https://iduarte.eu/">Isabel Duarte</a>, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '6eb557c1e1320049dc93c778f9efe639',
    indexName: 'quincunx',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
