<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Exploring PGS scores by Mavaddat et al. (2018) • quincunx</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Exploring PGS scores by Mavaddat et al. (2018)">
<meta property="og:description" content="quincunx">
<meta property="og:image" content="https://maialab.org/quincunx/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">quincunx</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    </li>
<li class="dropdown-header">PGS Catalog Concepts</li>
    <li>
      <a href="../articles/identifiers.html">Identifiers</a>
    </li>
    <li>
      <a href="../articles/pgs-construction-process.html">PGS Construction Process and Stages</a>
    </li>
    <li>
      <a href="../articles/cohorts-samples-sample-sets.html">Cohorts, Samples and Sample Sets</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Getting PGS Catalog Data</li>
    <li>
      <a href="../articles/getting-pgs-scores.html">Getting PGS Scores</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-publications.html">Getting PGS Publications</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-sample-sets.html">Getting PGS Sample Sets</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-performance-metrics.html">Getting PGS Performance Metrics</a>
    </li>
    <li>
      <a href="../articles/getting-traits.html">Getting Traits</a>
    </li>
    <li>
      <a href="../articles/getting-cohorts.html">Getting Cohorts</a>
    </li>
    <li>
      <a href="../articles/getting-releases.html">PGS Catalog Releases</a>
    </li>
    <li>
      <a href="../articles/pgs-scoring-file.html">PGS scoring file</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Wrangling PGS Catalog Data</li>
    <li>
      <a href="../articles/pgs-cat-ent-subsetting.html">PGS Catalog Entity Subsetting</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Case Studies</li>
    <li>
      <a href="../articles/risk-strat-brs-patients.html">Risk of Brugada syndrome patients</a>
    </li>
    <li>
      <a href="../articles/pgs-scores-mavaddat.html">Exploring PGS scores by Mavaddat et al. (2018)</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../articles/index.html">More articles...</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/maialab/quincunx/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pgs-scores-mavaddat_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Exploring PGS scores by Mavaddat et al. (2018)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/maialab/quincunx/blob/HEAD/vignettes/pgs-scores-mavaddat.Rmd" class="external-link"><code>vignettes/pgs-scores-mavaddat.Rmd</code></a></small>
      <div class="hidden name"><code>pgs-scores-mavaddat.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Let us assume that you are a breast cancer researcher, and that you are interested in studying the screening and prevention of this disease. Now, imagine you have just recently noticed a new publication that claims to have developed a set of new polygenic risk scores that are both powerful and reliable predictors of breast cancer risk <span class="citation">1</span>.</p>
<p>Perhaps now you would like to investigate a bit more about these new polygenic risk scores to assess their potential application. You know that the performance of these scores is dependent on various aspects, such as study design, participant demographics, case definitions, and covariates that have been adjusted for. In general, to access this information, you will have to carefully read the paper searching for these details, and usually get them from the supplementary material, with all the extra effort it takes.<br>
However, if these scores have already been indexed and manually curated by the <a href="https://www.pgscatalog.org/" class="external-link">PGS Catalog</a> team, then you can benefit by using this free and open resource to quickly gather relevant data about these scores. And if you happen to be an R user, then you can use the R Package <em>quincunx</em> and its retrieval functions to fetch the polygenic score metadata associated with the publication of interest from the PGS Catalog REST API server. This is what we will be doing next.</p>
</div>
<div class="section level2">
<h2 id="finding-polygenic-risk-scores-by-publication">Finding polygenic risk scores by publication<a class="anchor" aria-label="anchor" href="#finding-polygenic-risk-scores-by-publication"></a>
</h2>
<div class="section level3">
<h3 id="searching-for-publications-by-author-name">Searching for publications by author name<a class="anchor" aria-label="anchor" href="#searching-for-publications-by-author-name"></a>
</h3>
<p>To check if we are fortunate enough to have the polygenic risk scores for breast cancer <span class="citation">1</span> in the PGS Catalog, we start by loading the quincunx package and by searching for this publication in the PGS Catalog. To do this we will use the function <code><a href="../reference/get_publications.html">get_publications()</a></code> and start by searching by the first author’s last name, i.e. “Mavaddat”:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/maialab/quincunx" class="external-link">quincunx</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">pub_by_mavaddat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_publications.html">get_publications</a></span><span class="op">(</span>author <span class="op">=</span> <span class="st">'Mavaddat'</span><span class="op">)</span> <span class="co"># Not case sensitive,</span></span>
<span>                                                         <span class="co"># so "mavaddat" would</span></span>
<span>                                                         <span class="co"># have worked just fine.</span></span></code></pre></div>
<p>When you ask the PGS Catalog for publications, it returns an S4 <code>publications</code> object with two tables (slots): <code>publications</code> and <code>pgs_ids</code>. You can access these tables with the <code>@</code> operator.</p>
<p>The <code>publications</code> table (data frame) contains all the publications added to the PGS Catalog that contain an author whose last name is “Mavaddat”:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pub_by_mavaddat</span><span class="op">@</span><span class="va">publications</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span></span>
<span><span class="co">#&gt;   pgp_id    pubmed_id publication_date publication   title autho…¹ doi   authors</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;date&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGP000001 25855707  2015-04-08       J Natl Cance… Pred… Mavadd… 10.1… Mavadd…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGP000002 30554720  2018-12-13       Am J Hum Gen… Poly… Mavadd… 10.1… Mavadd…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGP000033 28376175  2017-07-01       J Natl Cance… Eval… Kuchen… 10.1… Kuchen…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGP000109 33022221  2020-10-05       Am J Hum Gen… Brea… Kramer… 10.1… Kramer…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGP000112 32737321  2020-07-31       Nat Commun    Euro… Ho WK   10.1… Ho WK,…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGP000117 32665703  2020-07-15       Genet Med     Poly… Barnes… 10.1… Barnes…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable name ¹​author_fullname</span></span></span></code></pre></div>
<p>Seemingly there are 6 publications indexed in the PGS Catalog. Each publication added to the PGS Catalog gets an unique identifier. This identifier is in the first column (<code>pgp_id</code>) of the <code>publications</code> table. The number of publications can be obtained by using the function <code><a href="../reference/n.html">n()</a></code> on the S4 object <code>pub_by_mavaddat</code>, or by asking directly the number of rows of the <code>publications</code> table:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">pub_by_mavaddat</span><span class="op">)</span> <span class="co"># Here we use quincunx::n() instead of n() to avoid namespace collisions with dplyr::n().</span></span>
<span><span class="co">#&gt; [1] 6</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pub_by_mavaddat</span><span class="op">@</span><span class="va">publications</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 6</span></span></code></pre></div>
<p>Moreover, we can see that Nasim Mavaddat is not the first author in all these publications because her name is not always present in the column <code>author_fullname</code> (which contains the name of the first author only). If you want to know all the authors you can access the column <code>authors</code> (contains a string of comma separated author names).</p>
<p>Now, to identify our publication of interest amongst these 6 publications, we can check, for example, the PubMed identifier (column <code>pubmed_id</code>), or check other information such as journal name (column <code>publication</code>), the publication title (column <code>title</code>) or the publication date (column <code>publication_date</code>). The most unambiguous approach is to use the PubMed identifier. The PubMed identifier is <code>"30554720"</code>. So the publication of interest corresponds to the one returned in row number 2 of the <code>publications</code> table:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">publication_row_index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">pub_by_mavaddat</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">pubmed_id</span> <span class="op">==</span> <span class="st">"30554720"</span><span class="op">)</span></span>
<span><span class="va">publication_row_index</span></span>
<span><span class="co">#&gt; [1] 2</span></span></code></pre></div>
<p>To focus now on this publication and filter out the others, we can subset the <code>publications</code> object <code>pub_by_mavaddat</code>, e.g., by position, and get a new object with both tables (<code>publications</code> and <code>pgs_ids</code>) filtered for only this publication:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018</span> <span class="op">&lt;-</span> <span class="va">pub_by_mavaddat</span><span class="op">[</span><span class="va">publication_row_index</span><span class="op">]</span></span>
<span><span class="va">mavaddat2018</span></span>
<span><span class="co">#&gt; An object of class "publications"</span></span>
<span><span class="co">#&gt; Slot "publications":</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 8</span></span></span>
<span><span class="co">#&gt;   pgp_id    pubmed_id publication_date publication   title autho…¹ doi   authors</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;date&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGP000002 30554720  2018-12-13       Am J Hum Gen… Poly… Mavadd… 10.1… Mavadd…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable name ¹​author_fullname</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "pgs_ids":</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 3</span></span></span>
<span><span class="co">#&gt;    pgp_id    pgs_id    stage   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGP000002 PGS000004 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGP000002 PGS000005 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGP000002 PGS000006 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGP000002 PGS000007 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGP000002 PGS000008 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGP000002 PGS000009 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGP000002 PGS000001 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGP000002 PGS000002 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGP000002 PGS000003 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGP000002 PGS000004 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PGP000002 PGS000005 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PGP000002 PGS000006 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> PGP000002 PGS000007 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> PGP000002 PGS000008 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> PGP000002 PGS000009 eval</span></span></code></pre></div>
<p>We see now that the <code>publications</code> table contains one entry only, and that the PGS Catalog identifier assigned to this publication is PGP000002:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">pgp_id</span></span>
<span><span class="co">#&gt; [1] "PGP000002"</span></span></code></pre></div>
<p>In the <code>pgs_ids</code> we see that there are several PGS scores associated with publication PGP000002. Besides listing the score identifiers (<code>pgs_id</code>), it also includes the <code>stage</code> column that annotates the polygenic score relative to the PGS construction process (check page 2 of the <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf" class="external-link">quincunx cheatsheet</a> for a visual illustration).</p>
</div>
<div class="section level3">
<h3 id="searching-for-publications-by-pubmed-id">Searching for publications by PubMed ID<a class="anchor" aria-label="anchor" href="#searching-for-publications-by-pubmed-id"></a>
</h3>
<p>An alternatively, albeit more direct, route to get this publication could have been to query for publications directly by the corresponding PubMed ID (30554720):</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pub_by_pmid_30554720</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_publications.html">get_publications</a></span><span class="op">(</span>pubmed_id <span class="op">=</span> <span class="st">'30554720'</span><span class="op">)</span></span>
<span><span class="va">pub_by_pmid_30554720</span></span>
<span><span class="co">#&gt; An object of class "publications"</span></span>
<span><span class="co">#&gt; Slot "publications":</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 8</span></span></span>
<span><span class="co">#&gt;   pgp_id    pubmed_id publication_date publication   title autho…¹ doi   authors</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;date&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGP000002 30554720  2018-12-13       Am J Hum Gen… Poly… Mavadd… 10.1… Mavadd…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable name ¹​author_fullname</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slot "pgs_ids":</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 3</span></span></span>
<span><span class="co">#&gt;    pgp_id    pgs_id    stage   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGP000002 PGS000004 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGP000002 PGS000005 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGP000002 PGS000006 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGP000002 PGS000007 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGP000002 PGS000008 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGP000002 PGS000009 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGP000002 PGS000001 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGP000002 PGS000002 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGP000002 PGS000003 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGP000002 PGS000004 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PGP000002 PGS000005 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PGP000002 PGS000006 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> PGP000002 PGS000007 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> PGP000002 PGS000008 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> PGP000002 PGS000009 eval</span></span></code></pre></div>
<p>To get an overview of the possible search criteria for <code>get_publications</code> you can use the help function within R.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">get_publications</span></span>
<span></span>
<span><span class="co"># or alternatively</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="st">"get_publications"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="from-publication-to-polygenic-risk-scores">From publication to polygenic risk scores<a class="anchor" aria-label="anchor" href="#from-publication-to-polygenic-risk-scores"></a>
</h2>
<p>Now that we have found that our publication of interest exists in the PGS Catalog, with identifier PGP000002, we can check now which polygenic risk scores are annotated in the Catalog. Polygenic scores (PGS) in the PGS Catalog are indexed by an unique accession identifier of the form: “PGS000000” (“PGS” followed by six digits).</p>
<p>To get all PGS identifiers associated with Mavaddat’s publication we turn to the second table <code>pgs_ids</code> that maps publication identifiers (PGP) to PGS identifiers:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pub_by_pmid_30554720</span><span class="op">@</span><span class="va">pgs_ids</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 3</span></span></span>
<span><span class="co">#&gt;    pgp_id    pgs_id    stage   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGP000002 PGS000004 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGP000002 PGS000005 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGP000002 PGS000006 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGP000002 PGS000007 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGP000002 PGS000008 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGP000002 PGS000009 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGP000002 PGS000001 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGP000002 PGS000002 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGP000002 PGS000003 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGP000002 PGS000004 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PGP000002 PGS000005 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PGP000002 PGS000006 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> PGP000002 PGS000007 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> PGP000002 PGS000008 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> PGP000002 PGS000009 eval</span></span></code></pre></div>
<p>Please note that there are <code>r length(unique(pub_by_pmid_30554720@pgs_ids$pgs_id))</code> unique scores, both from the <em>development</em> and the <em>evaluation</em> stages, meaning that this paper published new polygenic scores (<em>development</em> stage), and tested them (<em>evaluation</em> stage). But this paper has also evaluated 3 other polygenic scores previously developed (and firstly published in another publication by the same author). This distinction between stages is important because when we query the database for the scores from all the pgp_ids present in this publication, only the newly developed ones (from the <em>development stage</em>) will be retrieved. (See below: section about the <code><a href="../reference/get_scores.html">get_scores()</a></code> function).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Newly published PGS scores (development stage)</span></span>
<span><span class="va">pub_by_pmid_30554720</span><span class="op">@</span><span class="va">pgs_ids</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"development"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 3 variables: pgp_id &lt;chr&gt;, pgs_id &lt;chr&gt;, stage &lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span>
<span></span>
<span><span class="co"># All PGS scores evaluated (evaluation stage)  </span></span>
<span><span class="va">pub_by_pmid_30554720</span><span class="op">@</span><span class="va">pgs_ids</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">stage</span> <span class="op">==</span> <span class="st">"evaluation"</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 3 variables: pgp_id &lt;chr&gt;, pgs_id &lt;chr&gt;, stage &lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span></code></pre></div>
<p>If we knew, before hand, that PGP000002 was associated with Mavaddat’s publication, we could have also taken advantage of the neat function <code><a href="../reference/pgp_to_pgs.html">pgp_to_pgs()</a></code> to quickly get all the associated polygenic score ids:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/pgp_to_pgs.html">pgp_to_pgs</a></span><span class="op">(</span><span class="st">'PGP000002'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 3</span></span></span>
<span><span class="co">#&gt;    pgp_id    pgs_id    stage   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGP000002 PGS000004 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGP000002 PGS000005 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGP000002 PGS000006 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGP000002 PGS000007 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGP000002 PGS000008 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGP000002 PGS000009 gwas/dev</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGP000002 PGS000001 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGP000002 PGS000002 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGP000002 PGS000003 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGP000002 PGS000004 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PGP000002 PGS000005 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PGP000002 PGS000006 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> PGP000002 PGS000007 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> PGP000002 PGS000008 eval    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> PGP000002 PGS000009 eval</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="polygenic-scores-published-by-mavaddat-et-al--2018">Polygenic Scores published by Mavaddat <em>et al.</em> (2018)<a class="anchor" aria-label="anchor" href="#polygenic-scores-published-by-mavaddat-et-al--2018"></a>
</h2>
<p>To dive into the metadata about these polygenic scores, we use the quincunx function <code><a href="../reference/get_scores.html">get_scores()</a></code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_scores.html">get_scores</a></span><span class="op">(</span>pubmed_id <span class="op">=</span> <span class="st">"30554720"</span><span class="op">)</span></span>
<span><span class="fu">slotNames</span><span class="op">(</span><span class="va">mavaddat2018_scores</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "scores"                     "publications"              </span></span>
<span><span class="co">#&gt; [3] "samples"                    "demographics"              </span></span>
<span><span class="co">#&gt; [5] "cohorts"                    "traits"                    </span></span>
<span><span class="co">#&gt; [7] "stages_tally"               "ancestry_frequencies"      </span></span>
<span><span class="co">#&gt; [9] "multi_ancestry_composition"</span></span></code></pre></div>
<p>This returns the S4 object <code>scores</code> which contains six tables (slots): scores, publications, samples, demographics, cohorts, traits, stages_tally, ancestry_frequencies, multi_ancestry_composition.<br>
You can quickly check all variables from each table by consulting <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf" class="external-link">quincunx cheatsheet</a>.<br>
A description of each variable is annotated in the <code>scores</code> object help page that can be accessed with <code><a href="../reference/scores-class.html">class?scores</a></code>.</p>
</div>
<div class="section level2">
<h2 id="polygenic-scores-metadata">Polygenic Scores Metadata<a class="anchor" aria-label="anchor" href="#polygenic-scores-metadata"></a>
</h2>
<div class="section level3">
<h3 id="scores-object-scores-table">scores object | scores table<a class="anchor" aria-label="anchor" href="#scores-object-scores-table"></a>
</h3>
<p>The S4 scores object <code>mavaddat2018_scores</code> starts with a table named <code>scores</code> that lists each score in one row. All scores are identified by a PGS identifier (<code>pgs_id</code> column). Note that, as previously explained, only the polygenic scores newly developed in this publication (<em>developmental stage</em>) are retrieved, and not all the PGS scores that were evaluated in this publication.</p>
<p>In addition, scores can have a name (<code>pgs_name</code> column). This may be the name assigned by authors of the source publication, or a name assigned by a PGS Catalog curator in order to identify that particular score throughout the curating process.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'pgs_name'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   pgs_id    pgs_name     </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGS000004 PRS313_BC    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGS000005 PRS313_ERpos </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGS000006 PRS313_ERneg </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGS000007 PRS3820_BC   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGS000008 PRS3820_ERpos</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGS000009 PRS3820_ERneg</span></span></code></pre></div>
<p>From the PGS names we can already see the presence of the suffixes “ERpos” and “ERneg”, suggestive of specialized polygenic risk scores for estrogen-receptor positive and negative samples.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="st">'scoring_file'</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 1</span></span></span>
<span><span class="co">#&gt;   scoring_file                                                                  </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                                         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000004/ScoringFiles/PG…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000005/ScoringFiles/PG…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000006/ScoringFiles/PG…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000007/ScoringFiles/PG…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000008/ScoringFiles/PG…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> https://ftp.ebi.ac.uk/pub/databases/spot/pgs/scores/PGS000009/ScoringFiles/PG…</span></span></code></pre></div>
<p>The column <code>scoring_file</code> contains the URL for the FTP location containing the corresponding PGS scoring files. PGS scoring files are the text files provided by the PGS Catalog team containing the source data that you can use to compute polygenic scores for particular individuals, i.e. that allow you to apply these scores to your individual samples. Learn more about scoring files in <code><a href="../articles/pgs-scoring-file.html">vignette("pgs-scoring-file")</a></code>. For a quick consultation of the file format of PGS scoring files you may also check the second page of <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf" class="external-link">quincunx cheatsheet</a>.</p>
<p>As an additional feature, <em>quincunx</em> allows you to download the relevant PGS scoring files directly into R using the function <code><a href="../reference/read_scoring_file.html">read_scoring_file()</a></code>, making the data immediately available in R for further analysis.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'matches_publication'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   pgs_id    matches_publication</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGS000004 TRUE               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGS000005 TRUE               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGS000006 TRUE               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGS000007 TRUE               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGS000008 TRUE               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGS000009 TRUE</span></span></code></pre></div>
<p>The column <code>matches_publication</code> is a logical value indicating whether the published polygenic score is exactly the same as the one present in the PGS scoring file provided by the PGS Catalog. In this case all of the 6 scores are provided exactly as published (all values are “TRUE”).</p>
<p>Other columns in the <code>scores</code> table hold relevant information.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_name'</span>, <span class="st">'pgs_method_name'</span>, <span class="st">'pgs_method_params'</span>, <span class="st">'n_variants'</span>, <span class="st">'assembly'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span></span>
<span><span class="co">#&gt;   pgs_name      pgs_method_name                          pgs_m…¹ n_var…² assem…³</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PRS313_BC     Hard-Thresholding Stepwise Forward Regr… p &lt; 10…     313 GRCh37 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PRS313_ERpos  Hard-Thresholding Stepwise Forward Regr… p &lt; 10…     313 GRCh37 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PRS313_ERneg  Hard-Thresholding Stepwise Forward Regr… p &lt; 10…     313 GRCh37 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PRS3820_BC    LASSO penalized regression               p &lt; 0.…    <span style="text-decoration: underline;">3</span>820 GRCh37 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PRS3820_ERpos LASSO penalized regression               p &lt; 0.…    <span style="text-decoration: underline;">3</span>820 GRCh37 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PRS3820_ERneg LASSO penalized regression               p &lt; 0.…    <span style="text-decoration: underline;">3</span>820 GRCh37 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable names ¹​pgs_method_params, ²​n_variants, ³​assembly</span></span></span></code></pre></div>
<p>For example, columns such as <code>pgs_method_name</code> and <code>pgs_method_params</code> provide extra details about the PGS development method. Finally, <code>n_variants</code> informs about the number of variants comprising each polygenic risk score, and <code>assembly</code> indicates the genome assembly version used.</p>
</div>
<div class="section level3">
<h3 id="scores-object-publications-table">scores object | publications table<a class="anchor" aria-label="anchor" href="#scores-object-publications-table"></a>
</h3>
<p>The <code>scores</code> S4 object contains a table dedicated to the source publications used to collect the score(s) retrieved. In this case, it is not surprising that all PGS scores map to the same publication identifier, i.e., PGP000002, as that was our starting point.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">publications</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'pgp_id'</span>, <span class="st">'publication_date'</span>, <span class="st">'author_fullname'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   pgs_id    pgp_id    publication_date author_fullname</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;date&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGS000004 PGP000002 2018-12-13       Mavaddat N     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGS000005 PGP000002 2018-12-13       Mavaddat N     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGS000006 PGP000002 2018-12-13       Mavaddat N     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGS000007 PGP000002 2018-12-13       Mavaddat N     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGS000008 PGP000002 2018-12-13       Mavaddat N     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGS000009 PGP000002 2018-12-13       Mavaddat N</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="scores-object-samples-table">scores object | samples table<a class="anchor" aria-label="anchor" href="#scores-object-samples-table"></a>
</h3>
<p>The third table (slot) of the <code>scores</code> S4 object pertains to the samples used for the development of the PGS scores.</p>
<p>There are a total of 15 columns with metadata details about each sample. Each row corresponds to one sample associated with the polygenic scores, and the combination of values of the first two columns, <code>pgs_id</code> and <code>sample_id</code>, uniquely identifies each sample in this table. All samples shown in the <code>samples</code> table of a <code>scores</code> object are annotated with a <code>stage</code>, that can take two values: <code>"discovery"</code> or <code>"training"</code>.<br>
The <code>"discovery"</code> samples are typically used in determining the variants that are afterwards used for polygenic score development. These variants originate typically from Genome-Wide Association Studies (GWAS). Hence, these samples might be linked to the GWAS Catalog. If that is the case, this information is provided in the column <code>study_id</code>, indicating the GWAS Catalog accession identifier. You may find more information about these GWAS studies by using the function <code>gwasrapidd::get_studies()</code> from the <a href="https://rmagno.eu/gwasrapidd/" class="external-link">gwasrapidd</a> R package that we developed previously <span class="citation">2</span>.<br>
The <code>"training"</code> samples are those that have been used for the training of a particular polygenic score. Together, these two stages (<em>discovery</em> and <em>training</em>) are referred to as <em>development</em>, in contrast to the later testing phase of the polygenic scores, i.e., the <em>evaluation</em> phase (or stage).<br>
If this sounds confusing check our <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf" class="external-link">cheatsheet</a>, section <em>PGS Construction Process</em>, second page.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 12</span></span>
<span><span class="co">#&gt; Columns: 15</span></span>
<span><span class="co">#&gt; $ pgs_id                          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "PGS000004", "PGS000004", "PGS000005",…</span></span>
<span><span class="co">#&gt; $ sample_id                       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2</span></span>
<span><span class="co">#&gt; $ stage                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "gwas", "dev", "gwas", "dev", "gwas", …</span></span>
<span><span class="co">#&gt; $ sample_size                     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 158648, 10444, 87368, 5159, 87368, 515…</span></span>
<span><span class="co">#&gt; $ sample_cases                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 88916, 5159, 55391, 4233, 15404, 926, …</span></span>
<span><span class="co">#&gt; $ sample_controls                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 69732, 5285, 31977, 926, 71964, 4233, …</span></span>
<span><span class="co">#&gt; $ sample_percent_male             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0</span></span>
<span><span class="co">#&gt; $ phenotype_description           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Invasive breast cancer-affected", "In…</span></span>
<span><span class="co">#&gt; $ ancestry_category               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "European", "European", "European", "E…</span></span>
<span><span class="co">#&gt; $ ancestry                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ country                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ ancestry_additional_description <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ study_id                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ pubmed_id                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA…</span></span>
<span><span class="co">#&gt; $ cohorts_additional_description  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Training Cohort", "Validation Cohort"…</span></span></code></pre></div>
<p>The PGS Catalog provides brief records of samples sizes (total number, number of cases, and number of controls):</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 6</span></span></span>
<span><span class="co">#&gt;    pgs_id    sample_id stage sample_size sample_cases sample_controls</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGS000004         1 gwas       <span style="text-decoration: underline;">158</span>648        <span style="text-decoration: underline;">88</span>916           <span style="text-decoration: underline;">69</span>732</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGS000004         2 dev         <span style="text-decoration: underline;">10</span>444         <span style="text-decoration: underline;">5</span>159            <span style="text-decoration: underline;">5</span>285</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGS000005         1 gwas        <span style="text-decoration: underline;">87</span>368        <span style="text-decoration: underline;">55</span>391           <span style="text-decoration: underline;">31</span>977</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGS000005         2 dev          <span style="text-decoration: underline;">5</span>159         <span style="text-decoration: underline;">4</span>233             926</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGS000006         1 gwas        <span style="text-decoration: underline;">87</span>368        <span style="text-decoration: underline;">15</span>404           <span style="text-decoration: underline;">71</span>964</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGS000006         2 dev          <span style="text-decoration: underline;">5</span>159          926            <span style="text-decoration: underline;">4</span>233</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGS000007         1 gwas       <span style="text-decoration: underline;">158</span>648        <span style="text-decoration: underline;">88</span>916           <span style="text-decoration: underline;">69</span>732</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGS000007         2 dev         <span style="text-decoration: underline;">10</span>444         <span style="text-decoration: underline;">5</span>159            <span style="text-decoration: underline;">5</span>285</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGS000008         1 gwas        <span style="text-decoration: underline;">87</span>368        <span style="text-decoration: underline;">55</span>391           <span style="text-decoration: underline;">31</span>977</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGS000008         2 dev          <span style="text-decoration: underline;">5</span>159         <span style="text-decoration: underline;">4</span>233             926</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PGS000009         1 gwas        <span style="text-decoration: underline;">87</span>368        <span style="text-decoration: underline;">15</span>404           <span style="text-decoration: underline;">71</span>964</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PGS000009         2 dev          <span style="text-decoration: underline;">5</span>159          926            <span style="text-decoration: underline;">4</span>233</span></span></code></pre></div>
<p>Perhaps, not so surprisingly, the percentage of male participants in the training samples is zero:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'sample_id'</span>, <span class="st">'stage'</span>, <span class="st">'sample_percent_male'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 4</span></span></span>
<span><span class="co">#&gt;    pgs_id    sample_id stage sample_percent_male</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGS000004         1 gwas                    0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGS000004         2 dev                     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGS000005         1 gwas                    0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGS000005         2 dev                     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGS000006         1 gwas                    0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGS000006         2 dev                     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGS000007         1 gwas                    0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGS000007         2 dev                     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGS000008         1 gwas                    0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGS000008         2 dev                     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PGS000009         1 gwas                    0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PGS000009         2 dev                     0</span></span></code></pre></div>
<p>Also, information about the trait or disease studied and ancestry information can be accessed:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">samples</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'sample_id'</span>, <span class="st">'stage'</span>, <span class="st">'phenotype_description'</span>, <span class="st">'ancestry'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 5</span></span></span>
<span><span class="co">#&gt;    pgs_id    sample_id stage phenotype_description           ancestry</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGS000004         1 gwas  Invasive breast cancer-affected <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGS000004         2 dev   Invasive breast cancer-affected <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGS000005         1 gwas  ER-positive breast cancer cases <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGS000005         2 dev   ER-positive breast cancer cases <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGS000006         1 gwas  ER-negative breast cancer cases <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGS000006         2 dev   ER-negative breast cancer cases <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGS000007         1 gwas  Invasive breast cancer-affected <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGS000007         2 dev   Invasive breast cancer-affected <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGS000008         1 gwas  ER-positive breast cancer cases <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGS000008         2 dev   ER-positive breast cancer cases <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PGS000009         1 gwas  ER-negative breast cancer cases <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> PGS000009         2 dev   ER-negative breast cancer cases <span style="color: #BB0000;">NA</span></span></span></code></pre></div>
<p>Again, not so surprisingly, all samples are of European ancestry, a bias recognized by the research community. You can find more details about the ancestry variable in the <code>ancestry_categories</code> column. These categories have been defined within the NHGRI-EBI GWAS Catalog framework. We provide these ancestry nomenclature in <em>quincunx</em> as a separate dataset named <code>ancestry_categories</code>. See <code><a href="../reference/ancestry_categories.html">?ancestry_categories</a></code> more details. To quickly lookup the definition of the <em>European</em> ancestry:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Quick look at the ancestries definitions</span></span>
<span><span class="va">ancestry_categories</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 19 × 6</span></span></span>
<span><span class="co">#&gt;    ancestry_category                     ances…¹ ances…² ances…³ defin…⁴ examp…⁵</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Aboriginal Australian                 Additi… OTH     #999999 <span style="color: #949494;">"</span>Inclu… Martu …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> African American or Afro-Caribbean    African AFR     #FFD900 <span style="color: #949494;">"</span>Inclu… Africa…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> African unspecified                   African AFR     #FFD900 <span style="color: #949494;">"</span>Inclu… Africa…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Asian unspecified                     Additi… ASN     #B15928 <span style="color: #949494;">"</span>Inclu… Asian,…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Central Asian                         Additi… ASN     #B15928 <span style="color: #949494;">"</span>Inclu… Silk R…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> East Asian                            East A… EAS     #4DAF4A <span style="color: #949494;">"</span>Inclu… Chines…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> European                              Europe… EUR     #377EB8 <span style="color: #949494;">"</span>Inclu… Spanis…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Greater Middle Eastern (Middle Easte… Greate… GME     #00CED1 <span style="color: #949494;">"</span>Inclu… Tunisi…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Hispanic or Latin American            Hispan… AMR     #E41A1C <span style="color: #949494;">"</span>Inclu… Brazil…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Native American                       Additi… OTH     #999999 <span style="color: #949494;">"</span>Inclu… Pima I…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Not reported                          Ancest… NR      #BBBBBB <span style="color: #949494;">"</span>Inclu… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Oceanian                              Additi… OTH     #999999 <span style="color: #949494;">"</span>Inclu… Solomo…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> Other                                 Additi… OTH     #999999 <span style="color: #949494;">"</span>Inclu… Surina…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> Other admixed ancestry                Additi… OTH     #999999 <span style="color: #949494;">"</span>Inclu… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> South Asian                           South … SAS     #984EA3 <span style="color: #949494;">"</span>Inclu… Bangla…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> South East Asian                      Additi… ASN     #B15928 <span style="color: #949494;">"</span>Inclu… Thai, …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> Sub-Saharan African                   African AFR     #FFD900 <span style="color: #949494;">"</span>Inclu… Yoruba…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> Multi-Ancestry (including Europeans)  Multi-… MAE     #A6CEE3 <span style="color: #949494;">"</span>Combi… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> Multi-Ancestry (excluding Europeans)  Multi-… MAO     #FF7F00 <span style="color: #949494;">"</span>Combi… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable names ¹​ancestry_class, ²​ancestry_class_symbol,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ³​ancestry_class_colour, ⁴​definition, ⁵​examples</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="scores-object-demographics-table">scores object | demographics table<a class="anchor" aria-label="anchor" href="#scores-object-demographics-table"></a>
</h3>
<p>The demographics table usually lists demographic information about each sample. For this study this table is however empty, meaning that this information was either not available from Mavaddat’s publication, or not included in the PGS Catalog.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">demographics</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 11</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 11 variables: pgs_id &lt;chr&gt;, sample_id &lt;int&gt;, variable &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   estimate_type &lt;chr&gt;, estimate &lt;dbl&gt;, unit &lt;chr&gt;, variability_type &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variability &lt;dbl&gt;, interval_type &lt;chr&gt;, interval_lower &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   interval_upper &lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span></code></pre></div>
<p>Nevertheless, the demographics variables, when present, are follow-up time, and age of study participants.</p>
<p>If you want to confirm that <em>quincunx</em> is retrieving exactly the same info as provided by the PGS Catalog web interface, you can always check this by showing online the metadata for your PGP publication of interest using the function <code>open_in_pgs_catalog</code>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/open_in_pgs_catalog.html">open_in_pgs_catalog</a></span><span class="op">(</span><span class="st">'PGP000002'</span>, pgs_catalog_entity <span class="op">=</span> <span class="st">'pgp'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="scores-object-cohorts-table">scores object | cohorts table<a class="anchor" aria-label="anchor" href="#scores-object-cohorts-table"></a>
</h3>
<p>In the cohorts table you can check which cohorts are associated with which samples. Note that the unique identification of a sample is given by the combination of the values of the first two columns: <code>pgs_id</code>, and <code>sample_id</code>.<br>
To learn more about the meaning of cohorts for the PGS Catalog, visit our check our <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf" class="external-link">cheatsheet</a>, section <em>Cohorts, Samples and Sample Sets</em>, second page.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">cohorts</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 558 × 4</span></span></span>
<span><span class="co">#&gt;    pgs_id    sample_id cohort_symbol cohort_name                                </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGS000004         1 ABCFS         Australian Breast Cancer Family Study      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGS000004         1 MCCS          Melbourne Collaborative Cohort Study       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGS000004         1 HMBCS         Hannover-Minsk Breast Cancer Study         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGS000004         1 LMBC          Leuven Multidisciplinary Breast Centre     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGS000004         1 MTLGEBCS      Montreal Gene-Environment Breast Cancer St…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGS000004         1 CGPS          Copenhagen General Population Study        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGS000004         1 KBCP          Kuopio Breast Cancer Project               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGS000004         1 OBCS          Oulu Breast Cancer Study                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGS000004         1 CECILE        CECILE Breast Cancer Study                 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGS000004         1 BBCC          Bavarian Breast Cancer Cases and Controls  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 548 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="scores-object-traits-table">scores object | traits table<a class="anchor" aria-label="anchor" href="#scores-object-traits-table"></a>
</h3>
<p>Finally, in the traits table, you have access to the traits (phenotypes) associated with these polygenic scores. In this study, all scores indicate “breast cancer” or one of its subtypes (column <code>trait</code>):</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">traits</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'efo_id'</span>, <span class="st">'trait'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   pgs_id    efo_id      trait                                   </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGS000004 EFO_0000305 breast carcinoma                        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGS000005 EFO_1000649 estrogen-receptor positive breast cancer</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGS000006 EFO_1000650 estrogen-receptor negative breast cancer</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGS000007 EFO_0000305 breast carcinoma                        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGS000008 EFO_1000649 estrogen-receptor positive breast cancer</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGS000009 EFO_1000650 estrogen-receptor negative breast cancer</span></span></code></pre></div>
<p>Compared to the author-reported trait (column <code>reported_trait</code> from table <code>scores</code>), the trait description in this table follows the controlled vocabulary of an ontology, i.e., the <a href="https://www.ebi.ac.uk/efo/" class="external-link">Experimental Factor Ontology (EFO)</a>. This way, traits are described objectively. This is very useful for comparing trait data among different studies where different reported trait descriptions might have been used. For example, if you want now to know what other polygenic scores may be deposited in the PGS Catalog that also study breast cancer — namely, breast carcinoma, estrogen-receptor positive breast cancer, or estrogen-receptor negative breast cancer — then you could use their respective EFO identifiers (EFO_0000305, EFO_1000649, or EFO_1000650) with the function <code><a href="../reference/get_scores.html">get_scores()</a></code>:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores_bc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_scores.html">get_scores</a></span><span class="op">(</span>efo_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">traits</span><span class="op">[[</span><span class="st">'efo_id'</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">scores_bc</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 107</span></span></code></pre></div>
<p>So there are 107 scores present in the PGS Catalog. Included in this set are the 6 scores originating from <a href="https://doi.org/10.1016/j.ajhg.2018.11.002" class="external-link">Mavaddat et al. (2018)</a>. If we want to proceed with analysing these other scores without including those from the Mavaddat’s study, we could use the function <code><a href="../reference/setop.html">setdiff()</a></code> to remove them from the S4 <code>scores</code> object:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use quincunx::setdiff to avoid collision with dplyr::setdiff()</span></span>
<span><span class="va">bc_scores_not_mavaddat2018</span> <span class="op">&lt;-</span> <span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/setop.html">setdiff</a></span><span class="op">(</span><span class="va">scores_bc</span>, <span class="va">mavaddat2018_scores</span><span class="op">)</span></span>
<span><span class="fu">quincunx</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">bc_scores_not_mavaddat2018</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 101</span></span>
<span><span class="va">bc_scores_not_mavaddat2018</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'reported_trait'</span>,<span class="st">'n_variants'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 101 × 3</span></span></span>
<span><span class="co">#&gt;    pgs_id    reported_trait   n_variants</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGS000001 <span style="color: #949494;">"</span>Breast Cancer<span style="color: #949494;">"</span>          77</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PGS000015 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>        <span style="text-decoration: underline;">5</span>218</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PGS000028 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>          83</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PGS000029 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>          76</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PGS000045 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>          88</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PGS000050 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>          44</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PGS000051 <span style="color: #949494;">"</span>Breast Cancer <span style="color: #949494;">"</span>         67</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PGS000052 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>         161</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PGS000072 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>         187</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PGS000153 <span style="color: #949494;">"</span>Breast cancer<span style="color: #949494;">"</span>          66</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 91 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span></code></pre></div>
<p>For other set operations, check their documentation page: <code><a href="../reference/setop.html">union()</a></code>, <code><a href="../reference/setop.html">intersect()</a></code> and <code><a href="../reference/setop.html">setequal()</a></code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="performance-metrics">Performance Metrics<a class="anchor" aria-label="anchor" href="#performance-metrics"></a>
</h2>
<p>According to the PGS Catalog team <span class="citation">3</span>:</p>
<blockquote>
<p>“Performance Metrics assess the validity of a PGS in a Sample Set, independent of the samples used for score development. Common metrics include standardised effect sizes (odds/hazard ratios [OR/HR], and regression coefficients [Beta]), classification accuracy metrics (e.g. AUROC, C-index, AUPRC), but other relevant metrics (e.g. calibration [Chi-square]) can also be recorded. The covariates used in the model (most commonly age, sex, and genetic principal components (PCs) to account of population structure) are also linking to each set of metrics. Multiple PGS can be evaluated on the same Sample Set and further indexed as directly comparable Performance Metrics.”</p>
</blockquote>
<p>In this section we will learn how to retrieve details about the evaluation of the polygenic scores developed in Mavaddat <em>et al.</em> (2018). To do this, we start by querying the performance metrics for our scores of interest.</p>
<p>Please recall that the <code>pgs_ids</code> and <code>reported_traits</code> for the polygenic scores newly developed in Mavaddat’s publication are:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'pgs_id'</span>, <span class="st">'reported_trait'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   pgs_id    reported_trait           </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGS000004 Breast Cancer            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGS000005 ER-positive Breast Cancer</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGS000006 ER-negative Breast Cancer</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PGS000007 Breast Cancer            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PGS000008 ER-positive Breast Cancer</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PGS000009 ER-negative Breast Cancer</span></span></code></pre></div>
<p>So we use these PGS identifiers to query the PGS Catalog for performance metrics using the function <code><a href="../reference/get_performance_metrics.html">get_performance_metrics()</a></code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_performance_metrics.html">get_performance_metrics</a></span><span class="op">(</span>pgs_id <span class="op">=</span> <span class="va">mavaddat2018_scores</span><span class="op">@</span><span class="va">scores</span><span class="op">$</span><span class="va">pgs_id</span><span class="op">)</span></span></code></pre></div>
<p>The output is an S4 object with 9 tables:</p>
<ul>
<li>performance_metrics, publications, sample_sets, samples, demographics, cohorts, pgs_effect_sizes, pgs_classification_metrics, and pgs_other_metrics.</li>
</ul>
<p>Reminder | You can quickly check all variables from each table by consulting <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf" class="external-link">quincunx cheatsheet</a>.<br>
Also, a description of each variable is annotated in the <code>performance_metrics</code> object help page that can be accessed with <code><a href="../reference/performance_metrics-class.html">class?performance_metrics</a></code>.</p>
<div class="section level3">
<h3 id="performance_metrics-object-performance_metrics-table">performance_metrics object | performance_metrics table<a class="anchor" aria-label="anchor" href="#performance_metrics-object-performance_metrics-table"></a>
</h3>
<p>In the first table <code>performance_metrics</code> we get one performance metrics entity per row, with the following columns: ppm_id, pgs_id, reported_trait, covariates, comments.<br>
Note that all performance metrics are indexed with an unique identifier that starts with “PPM”.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">performance_metrics</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 143 × 4</span></span></span>
<span><span class="co">#&gt;    ppm_id    pgs_id    reported_trait                                    covar…¹</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PPM000004 PGS000004 Invasive breast cancer                            study,…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PPM000005 PGS000004 Incident breast cancer cases                      study,…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PPM000650 PGS000004 Breast cancer intrinsic-like subtype (luminal A-… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PPM000653 PGS000004 Breast cancer intrinsic-like subtype (luminal B/… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PPM000656 PGS000004 Breast cancer intrinsic-like subtype (luminal B-… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PPM000659 PGS000004 Breast cancer intrinsic-like subtype (HER2-enric… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PPM000662 PGS000004 Breast cancer intrinsic-like subtype (triple neg… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PPM000934 PGS000004 Metachronous contralateral breast cancer          Country</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PPM000935 PGS000004 Metachronous contralateral breast cancer          Countr…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PPM000936 PGS000004 Invasive metachronous contralateral breast cancer Country</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 133 more rows, and abbreviated variable name ¹​covariates</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span></code></pre></div>
<p>According to the PGS Catalog documentation (<a href="http://www.pgscatalog.org/docs/" class="external-link uri">http://www.pgscatalog.org/docs/</a>):</p>
<ul>
<li>The <code>reported_trait</code> displays the reported trait, often corresponding to the test set names reported in the publication, or more specific aspects of the phenotype being tested (e.g. if the disease cases are incident vs. recurrent events).<br>
</li>
<li>The <code>covariates</code> column lists the covariates used in the prediction model to evaluate the PGS. Examples include: age, sex, smoking habits, etc.<br>
</li>
<li>The <code>comments</code> column is a field where additional relevant information can be added to aid with understanding a particular performance metrics.</li>
</ul>
<p>Looking at the <code>performance_metrics</code> table, we can see that one polygenic score (<code>pgs_id</code>) can be associated with several performance metrics (<code>ppm_id</code>), e.g., PGS000007 associates with 8 PPMs:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">performance_metrics</span>, <span class="va">pgs_id</span> <span class="op">==</span> <span class="st">'PGS000007'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 8 × 5</span></span></span>
<span><span class="co">#&gt;   ppm_id    pgs_id    reported_trait                   covariates        comme…¹</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PPM000008 PGS000007 Invasive breast cancer           study, genetic P… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PPM000384 PGS000007 Breast Cancer (personal history) age at menarche   <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PPM000386 PGS000007 Breast Cancer (personal history) age, sex          <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PPM000388 PGS000007 Breast Cancer (personal history) <span style="color: #BB0000;">NA</span>                <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PPM005160 PGS000007 Breast cancer                    Age, family hist… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PPM005163 PGS000007 Breast cancer                    Age, family hist… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> PPM005169 PGS000007 Breast cancer                    <span style="color: #BB0000;">NA</span>                <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">8</span> PPM005172 PGS000007 Breast cancer                    <span style="color: #BB0000;">NA</span>                <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable name ¹​comments</span></span></span></code></pre></div>
<p>This means that the polygenic score (PGS000007) has been validated several times, using different models. In this case, we can immediately see that PGS000007 performance has been evaluated, for example, for alternative breast cancer types (different <code>reported_traits</code>), namely:</p>
<ul>
<li>Invasive breast cancer, Breast Cancer (personal history), Breast cancer.</li>
</ul>
<p>Additionally, we can see that the same <code>reported_trait</code> (<em>Breast Cancer (personal history)</em>) has been validated by 3 different performance metrics (PPMs): PPM000384, PPM000386, PPM000388; each having included different covariates in the model: age at menarche, age, sex, NA. (NA means that data for this field is <em>Not Available</em> in the records).</p>
</div>
<div class="section level3">
<h3 id="performance_metrics-object-publications-table">performance_metrics object | publications table<a class="anchor" aria-label="anchor" href="#performance_metrics-object-publications-table"></a>
</h3>
<p>The publications table is dedicated to hold information related to the publications associated with the performance metrics. The column <code>pgp_id</code> links each performance metrics to the respective publication where that performance metrics was reported and collected.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">publications</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 143 × 8</span></span></span>
<span><span class="co">#&gt;    ppm_id    pgp_id    pubmed_id publication_date publicat…¹ title autho…² doi  </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;date&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PPM000004 PGP000002 30554720  2018-12-13       Am J Hum … Poly… Mavadd… 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PPM000005 PGP000002 30554720  2018-12-13       Am J Hum … Poly… Mavadd… 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PPM000650 PGP000088 32424353  2020-05-18       Nat Genet  Geno… Zhang H 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PPM000653 PGP000088 32424353  2020-05-18       Nat Genet  Geno… Zhang H 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PPM000656 PGP000088 32424353  2020-05-18       Nat Genet  Geno… Zhang H 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PPM000659 PGP000088 32424353  2020-05-18       Nat Genet  Geno… Zhang H 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PPM000662 PGP000088 32424353  2020-05-18       Nat Genet  Geno… Zhang H 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PPM000934 PGP000109 33022221  2020-10-05       Am J Hum … Brea… Kramer… 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PPM000935 PGP000109 33022221  2020-10-05       Am J Hum … Brea… Kramer… 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PPM000936 PGP000109 33022221  2020-10-05       Am J Hum … Brea… Kramer… 10.1…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 133 more rows, and abbreviated variable names ¹​publication,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​author_fullname</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span></code></pre></div>
<p>Here we can immediately see that there are more publications than just the Mavaddat <em>et al.</em> (pubmed_id = 30554720) that we started with.<br>
This is expected because we requested all the performance metrics for the PGS scores that were newly <strong>developed</strong> by Mavaddat <em>et al.</em>; but these scores have been subsequently <strong>evaluated</strong> by other posterior publications, and accordingly have performance metrics reported in these posterior evaluations.<br>
This is easily confirmed by checking that all other publications are dated after December 13th 2018, which is the date of publication of the original Mavaddat <em>et al.</em> paper.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">publication_date</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "2018-12-13" "2020-05-18" "2020-10-05" "2020-07-15" "2020-12-14"</span></span>
<span><span class="co">#&gt;  [6] "2020-03-12" "2021-04-01" "2020-08-20" "2021-03-26" "2021-07-28"</span></span>
<span><span class="co">#&gt; [11] "2021-07-14" "2021-08-02" "2022-02-09" "2022-01-01" "2022-06-02"</span></span>
<span><span class="co">#&gt; [16] "2019-11-26"</span></span></code></pre></div>
<p>We can choose to look at those publications later to see what evaluations (performance metrics) were reported in them (for which traits, adjusting for which covariates, etc.).</p>
<p>But for now, we are only interested in studying the performance metrics reported by Mavaddat <em>et al.</em> for the PGS scores newly developed in this publication. So, let’s proceed with creating a vector containing only the <code>ppm_ids</code> of interest. We will do this by filtering the <code>ppm_ids</code> for the <code>pubmed_id</code> corresponding to the Mavaddat publication. We will then use this vector to subset the following tables (to display only the metrics for these PPMs of interest).</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm_ids</span> <span class="op">&lt;-</span> <span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">publications</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pubmed_id</span> <span class="op">==</span> <span class="st">"30554720"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">ppm_id</span><span class="op">)</span></span>
<span><span class="va">mavaddat2018_ppm_ids</span></span>
<span><span class="co">#&gt; [1] "PPM000004" "PPM000005" "PPM000006" "PPM000007" "PPM000008" "PPM000009"</span></span>
<span><span class="co">#&gt; [7] "PPM000010"</span></span>
<span></span>
<span><span class="co"># Find the corresponding pgs_id, reported_trait, covariates, and comments</span></span>
<span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">performance_metrics</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 5</span></span></span>
<span><span class="co">#&gt;   ppm_id    pgs_id    reported_trait               covariates            comme…¹</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PPM000004 PGS000004 Invasive breast cancer       study, genetic PCs 1… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PPM000005 PGS000004 Incident breast cancer cases study, genetic PCs 1… Includ…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PPM000006 PGS000005 ER-positive breast cancer    study, genetic PCs 1… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PPM000007 PGS000006 ER-negative breast cancer    study, genetic PCs 1… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PPM000008 PGS000007 Invasive breast cancer       study, genetic PCs 1… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PPM000009 PGS000008 ER-positive breast cancer    study, genetic PCs 1… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> PPM000010 PGS000009 ER-negative breast cancer    study, genetic PCs 1… <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable name ¹​comments</span></span></span></code></pre></div>
<p>Here, we can immediately see that the Mavaddat publication has evaluated the PGS000004 twice, with two performance metrics (PPM000004 and PPM000005), that are different because they include a different set of SNPs (see the <code>comments</code> column for PPM000005).</p>
</div>
<div class="section level3">
<h3 id="performance_metrics-object-sample_sets-table">performance_metrics object | sample_sets table<a class="anchor" aria-label="anchor" href="#performance_metrics-object-sample_sets-table"></a>
</h3>
<p>The PGS Catalog provides a Sample Set Id (PSS) that links the PPMs to the sample sets that were used to evaluate the corresponding PGS. This mapping is stored in the <code>sample_sets</code> table.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">sample_sets</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 2</span></span></span>
<span><span class="co">#&gt;   ppm_id    pss_id   </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PPM000004 PSS000004</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PPM000005 PSS000007</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PPM000006 PSS000005</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PPM000007 PSS000006</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PPM000008 PSS000004</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PPM000009 PSS000005</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> PPM000010 PSS000006</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="performance_metrics-object-samples-table">performance_metrics object | samples table<a class="anchor" aria-label="anchor" href="#performance_metrics-object-samples-table"></a>
</h3>
<p>The <code>samples</code> table gathers more relevant information regarding the samples used for the relevant evaluations. This table contains the following columns:</p>
<ul>
<li>ppm_id, pss_id, sample_id, stage, sample_size, sample_cases, sample_controls, sample_percent_male, phenotype_description, ancestry_category, ancestry, country, ancestry_additional_description, study_id, pubmed_id, cohorts_additional_description.</li>
</ul>
<p>Please note that the samples are not identified in PGS Catalog with a global unique identifier, but <em>quincunx</em> assigns a surrogate identifier (<code>sample_id</code>) to allow the mapping between tables.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">samples</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 16</span></span></span>
<span><span class="co">#&gt;   ppm_id    pss_id sampl…¹ stage sampl…² sampl…³ sampl…⁴ sampl…⁵ pheno…⁶ ances…⁷</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PPM000004 PSS00…       1 eval    <span style="text-decoration: underline;">29</span>751   <span style="text-decoration: underline;">11</span>428   <span style="text-decoration: underline;">18</span>323       0 Invasi… Europe…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PPM000005 PSS00…       1 eval   <span style="text-decoration: underline;">190</span>040    <span style="text-decoration: underline;">3</span>215  <span style="text-decoration: underline;">186</span>825       0 Incide… Europe…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PPM000006 PSS00…       1 eval    <span style="text-decoration: underline;">11</span>428    <span style="text-decoration: underline;">7</span>992    <span style="text-decoration: underline;">3</span>436       0 ER-pos… Europe…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PPM000007 PSS00…       1 eval    <span style="text-decoration: underline;">11</span>428    <span style="text-decoration: underline;">1</span>259   <span style="text-decoration: underline;">10</span>169       0 ER-neg… Europe…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PPM000008 PSS00…       1 eval    <span style="text-decoration: underline;">29</span>751   <span style="text-decoration: underline;">11</span>428   <span style="text-decoration: underline;">18</span>323       0 Invasi… Europe…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PPM000009 PSS00…       1 eval    <span style="text-decoration: underline;">11</span>428    <span style="text-decoration: underline;">7</span>992    <span style="text-decoration: underline;">3</span>436       0 ER-pos… Europe…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> PPM000010 PSS00…       1 eval    <span style="text-decoration: underline;">11</span>428    <span style="text-decoration: underline;">1</span>259   <span style="text-decoration: underline;">10</span>169       0 ER-neg… Europe…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 6 more variables: ancestry &lt;chr&gt;, country &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ancestry_additional_description &lt;chr&gt;, study_id &lt;chr&gt;, pubmed_id &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   cohorts_additional_description &lt;chr&gt;, and abbreviated variable names</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ¹​sample_id, ²​sample_size, ³​sample_cases, ⁴​sample_controls,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁵​sample_percent_male, ⁶​phenotype_description, ⁷​ancestry_category</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span></code></pre></div>
<p>Here we can, for example see that the sample sizes are very different for each evaluation. Let’s take a quick look at their values.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">samples</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ppm_id</span>, <span class="va">sample_cases</span>, <span class="va">sample_controls</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="op">!</span><span class="va">ppm_id</span>, names_to <span class="op">=</span> <span class="st">"sample_type"</span>, values_to <span class="op">=</span> <span class="st">"count"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">sample_type</span>, y<span class="op">=</span><span class="va">count</span>, x<span class="op">=</span><span class="va">ppm_id</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">20</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="pgs-scores-mavaddat_files/figure-html/ppm_samples_plot-1.png" width="672" style="display: block; margin: auto;"></p>
<p>This plot clearly shows that PPM000005 uses a very large sample (particularly regarding the number of controls) when compared with all other reported PPMs.</p>
</div>
<div class="section level3">
<h3 id="performance_metrics-object-demographics-table">performance_metrics object | demographics table<a class="anchor" aria-label="anchor" href="#performance_metrics-object-demographics-table"></a>
</h3>
<p>The <code>demographics</code> table holds information regarding the demographics’ variables of each <code>sample</code>. Each demographics’ variable (row) is uniquely identified by the combination of values from the columns: <code>ppm_id</code>, <code>pss_id</code>, <code>sample_id</code>, and <code>variable</code>. Currently, the PGS Catalog only describes two demographic variables: <em>age of participants</em> and <em>follow-up time</em>.</p>
<p>The columns presented in the table are:</p>
<ul>
<li>ppm_id, pss_id, sample_id, variable, estimate_type, estimate, unit, variability_type, variability, interval_type, interval_lower, interval_upper</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">demographics</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 17</span></span>
<span><span class="co">#&gt; Columns: 12</span></span>
<span><span class="co">#&gt; $ ppm_id           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "PPM001345", "PPM001347", "PPM012909", "PPM012910", "…</span></span>
<span><span class="co">#&gt; $ pss_id           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "PSS000450", "PSS000450", "PSS009609", "PSS009610", "…</span></span>
<span><span class="co">#&gt; $ sample_id        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1</span></span>
<span><span class="co">#&gt; $ variable         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "age", "age", "age", "age", "age", "age", "age", "age…</span></span>
<span><span class="co">#&gt; $ estimate_type    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "mean age (at the end of follow-up)", "mean age (at t…</span></span>
<span><span class="co">#&gt; $ estimate         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 58.5, 58.5, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, 4…</span></span>
<span><span class="co">#&gt; $ unit             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "years", "years", "years", "years", "years", "years",…</span></span>
<span><span class="co">#&gt; $ variability_type <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</span></span>
<span><span class="co">#&gt; $ variability      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…</span></span>
<span><span class="co">#&gt; $ interval_type    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "iqr", "iqr", "range", "range", "range", "range", "ra…</span></span>
<span><span class="co">#&gt; $ interval_lower   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 45.10, 45.10, 0.00, 50.00, 0.00, 50.00, 50.00, 50.00,…</span></span>
<span><span class="co">#&gt; $ interval_upper   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 72.20, 72.20, 50.00, 100.00, 50.00, 100.00, 70.00, 70…</span></span></code></pre></div>
<p>Here we can see that neither of the PPMs shown is from the Mavaddat paper. This means that the performance metrics reported in the paper do not have any information regarding the demographics’ variables that belong in this table (age and follow-up time).</p>
</div>
<div class="section level3">
<h3 id="performance_metrics-object-cohorts-table">performance_metrics object | cohorts table<a class="anchor" aria-label="anchor" href="#performance_metrics-object-cohorts-table"></a>
</h3>
<p>Similarly to the <code>cohorts</code> table described above (in the <code>scores</code> object class), this table shows which cohorts are associated with each sample. However here, the unique identification of a sample can be obtained by combining the values of the first three columns: <code>ppm_id</code>, <code>pss_id</code> and <code>sample_id</code>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">cohorts</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 61 × 5</span></span></span>
<span><span class="co">#&gt;    ppm_id    pss_id    sample_id cohort_symbol cohort_name                      </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PPM000004 PSS000004         1 AHS           Agricultural Health Study        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PPM000004 PSS000004         1 BGS           Breakthrough Generations Study   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PPM000004 PSS000004         1 EPIC          European Prospective Investigati…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PPM000004 PSS000004         1 FHRISK        Family History Clinic in Manches…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PPM000004 PSS000004         1 NHS           Nurses Health Study              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PPM000004 PSS000004         1 NHS2          Nurses Health Study II           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PPM000004 PSS000004         1 PLCO          Prostate, Lung, Colorectal, and …</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PPM000004 PSS000004         1 PROCAS        Breast Screening Programme (NHSB…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PPM000004 PSS000004         1 KARMA         Karolinska Mammography Project f…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PPM000004 PSS000004         1 SISTER        The Sister Study                 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 51 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="performance_metrics-object-pgs-effect-sizes-pgs-classification-metrics-pgs-other-metrics-tables">performance_metrics object | PGS effect sizes &amp; PGS classification metrics &amp; PGS other metrics tables<a class="anchor" aria-label="anchor" href="#performance_metrics-object-pgs-effect-sizes-pgs-classification-metrics-pgs-other-metrics-tables"></a>
</h3>
<p>The three final tables of the <code>performance_metrics</code> object hold the performance metrics themselves used in the validation. Each table presents the same column structure with 11 total columns, where the second column is different between the three tables. This column shows an id created by <em>quincunx</em> that is used to identify each of the individual metrics tables (<code>effect_size_id</code>, <code>classification_metrics_id</code>, or <code>other_metrics_id</code>). All other columns are for the same variables:</p>
<ul>
<li>ppm_id, effect_size_id, estimate_type_long, estimate_type, estimate, unit, variability_type, variability, interval_type, interval_lower, interval_upper.</li>
</ul>
<p>(These column names are specifically from the <code>pgs_effect_sizes</code> table, as indicated by the second column name. All other columns are equally named in all three tables).</p>
<p>According to the PGS Catalog online documentation (<a href="http://www.pgscatalog.org/docs/" class="external-link uri">http://www.pgscatalog.org/docs/</a>):</p>
<blockquote>
<p>“The reported values of the performance metrics are all reported similarly (e.g. the estimate is recorded along with the 95% confidence interval (if supplied)) and grouped according to the type of statistic they represent:<br>
- <strong>PGS Effect Sizes (per SD change)</strong> | Standardized effect sizes, per standard deviation [SD] change in PGS. Examples include regression coefficients (Betas) for continuous traits, Odds ratios (OR) and/or Hazard ratios (HR) for dichotomous traits depending on the availability of time-to-event data.<br>
- <strong>PGS Classification Metrics</strong> | Examples include the Area under the Receiver Operating Characteristic (AUROC) or Harrell’s C-index (Concordance statistic).<br>
- <strong>Other Metrics</strong> | Metrics that do not fit into the other two categories. Examples include: R2 (proportion of the variance explained), or reclassification metrics.”</p>
</blockquote>
<p>Now, let’s briefly explore the data in these tables (as usual filtered for only the PPMs that were newly reported in Mavaddat <em>et al.</em>).</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">pgs_effect_sizes</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 11</span></span></span>
<span><span class="co">#&gt;   ppm_id   effec…¹ estim…² estim…³ estim…⁴ unit  varia…⁵ varia…⁶ inter…⁷ inter…⁸</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PPM0000…       1 Odds R… OR         1.61 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci         1.57</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PPM0000…       1 Hazard… HR         1.59 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci         1.54</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PPM0000…       1 Odds R… OR         1.68 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci         1.63</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PPM0000…       1 Odds R… OR         1.45 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci         1.37</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PPM0000…       1 Odds R… OR         1.66 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci         1.61</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PPM0000…       1 Odds R… OR         1.73 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci         1.68</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> PPM0000…       1 Odds R… OR         1.44 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci         1.36</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 1 more variable: interval_upper &lt;dbl&gt;, and abbreviated variable names</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ¹​effect_size_id, ²​estimate_type_long, ³​estimate_type, ⁴​estimate,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁵​variability_type, ⁶​variability, ⁷​interval_type, ⁸​interval_lower</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span>
<span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">pgs_classification_metrics</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 11</span></span></span>
<span><span class="co">#&gt;   ppm_id   class…¹ estim…² estim…³ estim…⁴ unit  varia…⁵ varia…⁶ inter…⁷ inter…⁸</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PPM0000…       1 Area U… AUROC     0.63  <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PPM0000…       1 Area U… AUROC     0.641 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PPM0000…       1 Area U… AUROC     0.601 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> PPM0000…       1 Area U… AUROC     0.636 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PPM0000…       1 Area U… AUROC     0.647 <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> PPM0000…       1 Area U… AUROC     0.6   <span style="color: #BB0000;">NA</span>    se           <span style="color: #BB0000;">NA</span> ci           <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 1 more variable: interval_upper &lt;dbl&gt;, and abbreviated variable names</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ¹​classification_metrics_id, ²​estimate_type_long, ³​estimate_type, ⁴​estimate,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁵​variability_type, ⁶​variability, ⁷​interval_type, ⁸​interval_lower</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span>
<span><span class="va">mavaddat2018_ppm</span><span class="op">@</span><span class="va">pgs_other_metrics</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ppm_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mavaddat2018_ppm_ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 11</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 11 variables: ppm_id &lt;chr&gt;, other_metrics_id &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   estimate_type_long &lt;chr&gt;, estimate_type &lt;chr&gt;, estimate &lt;dbl&gt;, unit &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variability_type &lt;chr&gt;, variability &lt;dbl&gt;, interval_type &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   interval_lower &lt;dbl&gt;, interval_upper &lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ Use `colnames()` to see all variable names</span></span></span></code></pre></div>
<p>We can immediately see that the third table (reserved for metrics other than effect sizes and classification metrics) is empty; and that the effect sizes were estimated using Odds Ratio and Hazard Ratio, and the classification metric applied was AUROC.</p>
<p>Now we can look at the reported values for each PPM metrics and decide if the validation is relevant, and therefore make an informed choice to use the associated PGS score for our own study, and eventually apply it to our own dataset.</p>
</div>
</div>
<div class="section level2">
<h2 id="concluding-remarks">Concluding remarks<a class="anchor" aria-label="anchor" href="#concluding-remarks"></a>
</h2>
<p>Useful reminders:</p>
<ul>
<li>You can quickly check all variables from each table by consulting <a href="https://github.com/ramiromagno/cheatsheets/blob/master/quincunx/quincunx_cheatsheet.pdf" class="external-link">quincunx cheatsheet</a>.<br>
</li>
<li>All <em>quincunx</em> functions are annotated with help pages accessed with <code>?function_name</code> (e.g. <code><a href="../reference/get_scores.html">?get_scores</a></code>, <code><a href="../reference/get_traits.html">?get_traits</a></code>, <code><a href="../reference/get_performance_metrics.html">?get_performance_metrics</a></code>).<br>
</li>
<li>All tables retrieved by <em>quincunx’s</em> functions are annotated with help files that describe all variables present. These are accessible via <code>class?object_name</code> (e.g. <code><a href="../reference/scores-class.html">class?scores</a></code>, <code><a href="../reference/performance_metrics-class.html">class?performance_metrics</a></code>).</li>
<li>The <em>quincunx</em> package has been published in a peer-reviewed journal. Use <code>citation(package="quincunx")</code> to get the full paper citation.</li>
</ul>
</div>
<div class="section level2 unnumbered">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references">
<div id="ref-mavaddat2019">
<p>1. Mavaddat, N. <em>et al.</em> Polygenic Risk Scores for Prediction of Breast Cancer and Breast Cancer Subtypes. <em>The American Journal of Human Genetics</em> <strong>104</strong>, 21–34 (2019).</p>
</div>
<div id="ref-Magno.B.2019">
<p>2. Magno, R. &amp; Maia, A.-T. Gwasrapidd: An R package to query, download and wrangle GWAS catalog data. <em>Bioinformatics</em> (2019) doi:<a href="https://doi.org/10.1093/bioinformatics/btz605" class="external-link">10.1093/bioinformatics/btz605</a>.</p>
</div>
<div id="ref-Lambert.NG.2021">
<p>3. Lambert, S. A. <em>et al.</em> The Polygenic Score Catalog as an open database for reproducibility and systematic evaluation. <em>Nature Genetics</em> <strong>53</strong>, 420–425 (2021).</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, <a href="http://iduarte.eu/" class="external-link">Isabel Duarte</a>, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '6eb557c1e1320049dc93c778f9efe639',
    indexName: 'quincunx',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
