<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Risk of Brugada syndrome patients • quincunx</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Risk of Brugada syndrome patients">
<meta property="og:description" content="quincunx">
<meta property="og:image" content="https://maialab.org/quincunx/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">quincunx</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    <li class="dropdown-header">PGS Catalog Concepts</li>
    <li>
      <a href="../articles/identifiers.html">Identifiers</a>
    </li>
    <li>
      <a href="../articles/pgs-construction-process.html">PGS Construction Process</a>
    </li>
    <li>
      <a href="../articles/cohorts-samples-sample-sets.html">Cohorts, Samples and Sample Sets</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Getting PGS Catalog Data</li>
    <li>
      <a href="../articles/getting-pgs-scores.html">Getting PGS Scores</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-publications.html">Getting PGS Publications</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-sample-sets.html">Getting PGS Sample Sets</a>
    </li>
    <li>
      <a href="../articles/getting-pgs-performance-metrics.html">Getting PGS Performance Metrics</a>
    </li>
    <li>
      <a href="../articles/getting-traits.html">Getting Traits</a>
    </li>
    <li>
      <a href="../articles/getting-cohorts.html">Getting Cohorts</a>
    </li>
    <li>
      <a href="../articles/getting-releases.html">PGS Catalog Releases</a>
    </li>
    <li>
      <a href="../articles/pgs-scoring-file.html">PGS scoring file</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Wrangling PGS Catalog Data</li>
    <li>
      <a href="../articles/pgs-cat-ent-subsetting.html">PGS Catalog Entity Subsetting</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Case Studies</li>
    <li>
      <a href="../articles/risk-strat-brs-patients.html">Risk of Brugada syndrome patients</a>
    </li>
    <li>
      <a href="../articles/pgs-scores-mavaddat.html">Exploring PGS scores by Mavaddat et al. (2018)</a>
    </li>
    <li class="divider">
    <li>
      <a href="../articles/index.html">More...</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="risk-strat-brs-patients_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Risk of Brugada syndrome patients</h1>
            
      
      
      <div class="hidden name"><code>risk-strat-brs-patients.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>In this example we walk you through an hypothetical usage case of a quincunx application with the goal of calculating the risk of developing a specific disease. For illustrative purposes, the disease chosen here is the Brugada syndrome (for details about the origin of the name, see <a href="https://www.ecrjournal.com/articles/featuring-josep-brugada">Featuring: Josep Brugada</a>).</p>
<p>Brugada syndrome is a rare, inherited cardiac disease leading to ventricular fibrillation and sudden cardiac death in structurally normal hearts. People with Brugada syndrome have an increased risk of having irregular heart rhythms beginning in the lower chambers of the heart (ventricles) (<a href="https://doi.org/10.1038/gim.2015.35">10.1038/gim.2015.35</a>).</p>
<p>The prevalence varies among regions and ethnicities, affecting mostly males. The risk stratification and management of patients, predominantly asymptomatic, still remains challenging. Currently, despite several genes identified, SCN5A has attracted the most attention, and in approximately 30% of patients, a genetic variant may be implicated as causal factor after a comprehensive analysis (<a href="https://doi.org/10.1016/j.jacc.2018.06.037">10.1016/j.jacc.2018.06.037</a>).</p>
<p>In the walk-through below, we show how to use quincunx to search for Brugada syndrome related polygenic scores, and how to compute the associated risk depending on the risk alleles of the associated genetic variants.</p>
</div>
<div id="searching-for-brugada-syndrome-in-the-pgs-catalog" class="section level2">
<h2 class="hasAnchor">
<a href="#searching-for-brugada-syndrome-in-the-pgs-catalog" class="anchor"></a>Searching for Brugada syndrome in the PGS Catalog</h2>
<p>Before anything, let us load quincunx:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">quincunx</span><span class="op">)</span></code></pre></div>
<p>Now, we start by searching for the Brugada syndrome in the PGS Catalog. To do this we search for any traits/diseases that might include the term <code>"brugada"</code> in the trait description using the function <code><a href="../reference/get_traits.html">get_traits()</a></code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Search in PGS Catalog for traits/diseases containing the term "brugada"</span>
<span class="va">brs_traits</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_traits.html">get_traits</a></span><span class="op">(</span>trait_term <span class="op">=</span> <span class="st">'brugada'</span>, exact_term <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>We can see that indeed there is one trait specifically associated with the Brugada syndrome, and that it has the EFO trait Orphanet_130, whose description confirms it to be a <em>genetic disease ventricular arrhythmia that may result in sudden death</em>. In addition, from the table <code>pgs_ids</code> we can retrieve the associated PGS score identifiers. To date, only one polygenic score is associated with the Brugada syndrome: PGS000737.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># EFO Id</span>
<span class="va">brs_traits</span><span class="op">@</span><span class="va">traits</span><span class="op">$</span><span class="va">efo_id</span>
<span class="co">#&gt; [1] "Orphanet_130"</span>
<span class="co"># Brugada syndrome description as defined by the Experimental Factor Ontology</span>
<span class="va">brs_traits</span><span class="op">@</span><span class="va">traits</span><span class="op">$</span><span class="va">description</span>
<span class="co">#&gt; [1] "A genetically heterogeneous condition characterized by complete or incomplete right bundle branch block accompanied by ST elevation in leads V1-V3. There is a high incidence of ventricular arrhythmia that may result in sudden death. [NCIT: C142891]"</span>
<span class="co"># PGS scores associated with Brugada syndrome</span>
<span class="va">brs_traits</span><span class="op">@</span><span class="va">pgs_ids</span><span class="op">$</span><span class="va">pgs_id</span>
<span class="co">#&gt; [1] "PGS000737"</span></code></pre></div>
</div>
<div id="getting-details-about-the-polygenic-score-pgs000737" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-details-about-the-polygenic-score-pgs000737" class="anchor"></a>Getting details about the polygenic score PGS000737</h2>
<p>From the main table, <code>scores</code>, in <code>brs_pgs</code> object, we can see the following details:</p>
<ul>
<li>The name given to the polygenic score PGS000737 is <em>PRS_BrS</em>
</li>
<li>
<em>PRS_BrS</em> definition is in the PGS Catalog the same as the one in the original publication, because <code>matches_publication</code> is <code>TRUE</code>
</li>
<li>The original authors’ reported trait was <em>Brugada syndrome</em>
</li>
<li>And finally, the number of variants comprising <em>PRS_BrS</em> is three.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">brs_pgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_scores.html">get_scores</a></span><span class="op">(</span>pgs_id <span class="op">=</span> <span class="st">'PGS000737'</span><span class="op">)</span>

<span class="co"># PGS code name</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">scores</span><span class="op">$</span><span class="va">pgs_name</span>
<span class="co">#&gt; [1] "PRS_BrS"</span>
<span class="co"># Does the PGS score matches the one in the source publication?</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">scores</span><span class="op">$</span><span class="va">matches_publication</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="co"># Authors' own description of the trait</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">scores</span><span class="op">$</span><span class="va">reported_trait</span>
<span class="co">#&gt; [1] "Brugada syndrome"</span>
<span class="co"># Number of variants included in the polygenic risk score</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">scores</span><span class="op">$</span><span class="va">n_variants</span>
<span class="co">#&gt; [1] 3</span></code></pre></div>
</div>
<div id="original-publication-that-developed-pgs000737" class="section level2">
<h2 class="hasAnchor">
<a href="#original-publication-that-developed-pgs000737" class="anchor"></a>Original publication that developed PGS000737</h2>
<p>From the table <code>publications</code>, in <code>brs_pgs</code> object, we can see, <em>inter alia</em>, the following details about the publication behind the PGS PGS000737:</p>
<ul>
<li>The Catalog identifier assigned to this publication is PGP000144</li>
<li>The publication PubMed id is 31504448</li>
<li>It has been published in 2019-10-01</li>
<li>The publication title is: <em>Predicting cardiac electrical response to sodium-channel blockade and Brugada syndrome using polygenic risk scores.</em>
</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PGP id</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">pgp_id</span>
<span class="co">#&gt; [1] "PGP000144"</span>
<span class="co"># PubMed id</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">pubmed_id</span>
<span class="co">#&gt; [1] "31504448"</span>
<span class="co"># Publication date</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">publication_date</span>
<span class="co">#&gt; [1] "2019-10-01"</span>
<span class="co"># Journal</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">publication</span>
<span class="co">#&gt; [1] "Eur Heart J"</span>
<span class="co"># Title</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">title</span>
<span class="co">#&gt; [1] "Predicting cardiac electrical response to sodium-channel blockade and Brugada syndrome using polygenic risk scores."</span></code></pre></div>
<p>To view this publication online: <code><a href="../reference/open_in_pubmed.html">open_in_pubmed(brs_pgs@publications$pubmed_id)</a></code>.</p>
</div>
<div id="sample-ancestry" class="section level2">
<h2 class="hasAnchor">
<a href="#sample-ancestry" class="anchor"></a>Sample ancestry</h2>
<p>To get a feeling for the applicability of this polygenic score, it is important to know the genetic diversity of the participants involved in the development of the risk score. We can get details on the ancestry composition of the samples used to develop PRS_BrS by looking into the <code>samples</code> table. In this table we can see that one sample (<code>sample_id = 1</code>) of 1427 individuals was used at the gwas stage. The ancestry category associated with these individuals is European. You can use quincunx’s internal dataset on possible ancestry categories: <code>ancestry_categories</code>.</p>
<p>From this data, one may assume that the applicability of this risk score should probably be restricted to people from European ancestry.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Stage</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">samples</span><span class="op">$</span><span class="va">stage</span>
<span class="co">#&gt; [1] "gwas"</span>
<span class="co"># Sample size</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">samples</span><span class="op">$</span><span class="va">sample_size</span>
<span class="co">#&gt; [1] 1427</span>
<span class="co"># Ancestry</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">samples</span><span class="op">$</span><span class="va">ancestry_category</span>
<span class="co">#&gt; [1] "European"</span>
<span class="co"># Definition of European ancestry</span>
<span class="va">ancestry_categories</span><span class="op">[</span><span class="va">ancestry_categories</span><span class="op">$</span><span class="va">ancestry_category</span> <span class="op">==</span> <span class="st">'European'</span>, <span class="st">'definition'</span>, drop <span class="op">=</span> <span class="cn">TRUE</span><span class="op">]</span>
<span class="co">#&gt; [1] "Includes individuals who either self-report or have been described by authors as European, Caucasian, white, or one of the sub-populations from this region (e.g., Dutch). This category also includes individuals who genetically cluster with reference populations from this region, for example, 1000 Genomes and/or HapMap CEU, FIN, GBR, IBS, and TSI populations"</span>
<span class="co"># Cohort</span>
<span class="va">brs_pgs</span><span class="op">@</span><span class="va">cohorts</span>
<span class="co">#&gt; # A tibble: 1 × 4</span>
<span class="co">#&gt;   pgs_id    sample_id cohort_symbol cohort_name                                 </span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;         &lt;chr&gt;                                       </span>
<span class="co">#&gt; 1 PGS000737         1 DESIR         Epidemiological Study on the Insulin Resist…</span></code></pre></div>
</div>
<div id="the-genetic-composition-of-prs_brs" class="section level2">
<h2 class="hasAnchor">
<a href="#the-genetic-composition-of-prs_brs" class="anchor"></a>The genetic composition of PRS_BrS</h2>
<p>To get the polygenic risk score <em>PRS_BrS</em> (i.e., PGS000737), we start by downloading it using the function <code>read_scoring_file</code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PRS_BrS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_scoring_file.html">read_scoring_file</a></span><span class="op">(</span><span class="st">'PGS000737'</span><span class="op">)</span>
<span class="va">PRS_BrS</span>
<span class="co">#&gt; $PGS000737</span>
<span class="co">#&gt; $PGS000737$metadata</span>
<span class="co">#&gt; # A tibble: 1 × 7</span>
<span class="co">#&gt;   pgs_id    reported_trait  original_genome… number_of_varia… pgp_id citation   </span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;                       &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;      </span>
<span class="co">#&gt; 1 PGS000737 Brugada syndro… GRCh37                          3 PGP00… Tadros R e…</span>
<span class="co">#&gt; # … with 1 more variable: license &lt;chr&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $PGS000737$data</span>
<span class="co">#&gt; # A tibble: 3 × 8</span>
<span class="co">#&gt;   rsID       chr_name chr_position effect_allele reference_allele effect_weight</span>
<span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;           &lt;int&gt; &lt;chr&gt;         &lt;chr&gt;                    &lt;dbl&gt;</span>
<span class="co">#&gt; 1 rs11708996 3            38633923 C             G                         0.55</span>
<span class="co">#&gt; 2 rs10428132 3            38777554 G             T                        -0.94</span>
<span class="co">#&gt; 3 rs9388451  6           126090377 C             T                         0.46</span>
<span class="co">#&gt; # … with 2 more variables: weight_type &lt;chr&gt;, locus_name &lt;chr&gt;</span></code></pre></div>
<p>This PGS is comprised of only three variants: rs11708996, rs10428132, and rs9388451. These variants are located near loci well-know to be associated with the Brugada syndrome: SCN5A, SCN10A, NCOA7, and HEY2. From the <code>effect_weight</code> column, we can see that the effect allele (<code>"C"</code> in both cases) of variants rs11708996 and rs9388451 increases the risk score (<span class="math inline">\(\beta = 0.55\)</span> and <span class="math inline">\(\beta = 0.46\)</span>, respectively), whereas the <code>"G"</code> allele of rs10428132 has a protective effect relative to the <code>"T"</code> allele (<span class="math inline">\(\beta = -0.94\)</span>).</p>
<p>Now, to illustrate the calculation of <em>PRS_BrS</em> using appropriate patient-level data, we will generate all possible genotypes for these three variants, and calculate their respective risk scores. (We are not using real patient data here since such human datasets are not open). Each biallelic variant has three possible genotypes: A/A, A/a or a/a. Given that the score is also comprised of three variants, the total number of possible genotypes is <span class="math inline">\(3\times3\times3= 27\)</span>.</p>
<p>The function <code>genotypes</code> defined below generates the three genotypes for the given variant alleles:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genotypes</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">allele1</span>, <span class="va">allele2</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">alleles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">allele1</span>, <span class="va">allele2</span><span class="op">)</span>
  <span class="va">m</span> <span class="op">&lt;-</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/expand_grid.html">expand_grid</a></span><span class="op">(</span>allele1 <span class="op">=</span> <span class="va">alleles</span>, allele2 <span class="op">=</span> <span class="va">alleles</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">3</span>,<span class="op">]</span>
  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">m</span><span class="op">[</span>, <span class="fl">1</span>, drop <span class="op">=</span> <span class="cn">TRUE</span><span class="op">]</span>, <span class="va">m</span><span class="op">[</span>, <span class="fl">2</span>, drop <span class="op">=</span> <span class="cn">TRUE</span><span class="op">]</span>, sep <span class="op">=</span> <span class="st">'/'</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>For example, rs11708996 genotypes are:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">genotypes</span><span class="op">(</span><span class="st">'G'</span>, <span class="st">'C'</span><span class="op">)</span>
<span class="co">#&gt; [1] "G/G" "G/C" "C/C"</span></code></pre></div>
<p>The set of all 27 genotypes can be generated as follows:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">BrS_genotypes</span> <span class="op">&lt;-</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/expand_grid.html">expand_grid</a></span><span class="op">(</span>
  rs11708996 <span class="op">=</span> <span class="fu">genotypes</span><span class="op">(</span><span class="st">'G'</span>, <span class="st">'C'</span><span class="op">)</span>,
  rs10428132 <span class="op">=</span> <span class="fu">genotypes</span><span class="op">(</span><span class="st">'T'</span>, <span class="st">'G'</span><span class="op">)</span>,
  rs9388451 <span class="op">=</span> <span class="fu">genotypes</span><span class="op">(</span><span class="st">'T'</span>, <span class="st">'C'</span><span class="op">)</span>
<span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rowid_to_column</a></span><span class="op">(</span><span class="st">'patient'</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">BrS_genotypes</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 27 × 4</span>
<span class="co">#&gt;    patient rs11708996 rs10428132 rs9388451</span>
<span class="co">#&gt;      &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;    </span>
<span class="co">#&gt;  1       1 G/G        T/T        T/T      </span>
<span class="co">#&gt;  2       2 G/G        T/T        T/C      </span>
<span class="co">#&gt;  3       3 G/G        T/T        C/C      </span>
<span class="co">#&gt;  4       4 G/G        T/G        T/T      </span>
<span class="co">#&gt;  5       5 G/G        T/G        T/C      </span>
<span class="co">#&gt;  6       6 G/G        T/G        C/C      </span>
<span class="co">#&gt;  7       7 G/G        G/G        T/T      </span>
<span class="co">#&gt;  8       8 G/G        G/G        T/C      </span>
<span class="co">#&gt;  9       9 G/G        G/G        C/C      </span>
<span class="co">#&gt; 10      10 G/C        T/T        T/T      </span>
<span class="co">#&gt; 11      11 G/C        T/T        T/C      </span>
<span class="co">#&gt; 12      12 G/C        T/T        C/C      </span>
<span class="co">#&gt; 13      13 G/C        T/G        T/T      </span>
<span class="co">#&gt; 14      14 G/C        T/G        T/C      </span>
<span class="co">#&gt; 15      15 G/C        T/G        C/C      </span>
<span class="co">#&gt; 16      16 G/C        G/G        T/T      </span>
<span class="co">#&gt; 17      17 G/C        G/G        T/C      </span>
<span class="co">#&gt; 18      18 G/C        G/G        C/C      </span>
<span class="co">#&gt; 19      19 C/C        T/T        T/T      </span>
<span class="co">#&gt; 20      20 C/C        T/T        T/C      </span>
<span class="co">#&gt; 21      21 C/C        T/T        C/C      </span>
<span class="co">#&gt; 22      22 C/C        T/G        T/T      </span>
<span class="co">#&gt; 23      23 C/C        T/G        T/C      </span>
<span class="co">#&gt; 24      24 C/C        T/G        C/C      </span>
<span class="co">#&gt; 25      25 C/C        G/G        T/T      </span>
<span class="co">#&gt; 26      26 C/C        G/G        T/C      </span>
<span class="co">#&gt; 27      27 C/C        G/G        C/C</span></code></pre></div>
<p>In <code>BrS_genotypes</code> we have now a table of all possible genotypes comprising the three variants of <em>PRS_BrS</em>. We have prepended a column with a dummy patient identifier (<code>patient</code>) to refer to each one of the 27 possible genotypes in an unambiguous way.</p>
</div>
<div id="prs_brs-calculation" class="section level2">
<h2 class="hasAnchor">
<a href="#prs_brs-calculation" class="anchor"></a>PRS_BrS calculation</h2>
<p>If we assume an additive genetic architecture, with independence of risk variants, then the calculation of PRS_BrS is straightforward: simply multiply the number of effect alleles in the genotype by the <code>effect_weight</code>, and add up everything:</p>
<p><span class="math display">\[{PRS_{BrS}} = \beta_{\text{rs11708996}} \times x_{rs11708996} + \beta_{\text{rs10428132}} \times x_{rs10428132} + \beta_{\text{rs9388451}} \times x_{rs9388451}\]</span> where, e.g., <span class="math inline">\(\beta_{\text{rs11708996}} = 0.55\)</span>, and <span class="math inline">\(x_{rs11708996}\)</span> is either <span class="math inline">\(0\)</span>, for genotype “G/G”, <span class="math inline">\(1\)</span>, for genotype “G/C” or <span class="math inline">\(2\)</span>, for genotype “C/C”.</p>
<p>As a side note, although simplistic, the additive model reflects our best estimate of genetic architecture of common complex disorders, where little evidence of interaction between genetic variants is detected. And although this type of model ignores any gene-gene or gene-environment interactions (<a href="https://doi.org/10.1002/gepi.21989">10.1002/gepi.21989</a>), it seems, nevertheless, that the largest meta-analysis of heritability from twin studies supports a simple additive model in most of the traits examined (<a href="https://doi.org/10.1038/ng.3285">10.1038/ng.3285</a>).</p>
<p>Next we calculate PRS_BrS for the genotypes <code>BrS_genotypes</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">prs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Vectorize.html">Vectorize</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">scoring</span>, <span class="va">variant_id</span>, <span class="va">genotype</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">alleles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">genotype</span>, <span class="st">'/'</span><span class="op">)</span><span class="op">)</span>
  <span class="va">eff</span> <span class="op">&lt;-</span>
    <span class="va">scoring</span><span class="op">[</span><span class="va">scoring</span><span class="op">$</span><span class="va">rsID</span> <span class="op">==</span> <span class="va">variant_id</span>, <span class="st">'effect_allele'</span>, drop <span class="op">=</span> <span class="cn">TRUE</span><span class="op">]</span>
  <span class="va">weight</span> <span class="op">&lt;-</span>
    <span class="va">scoring</span><span class="op">[</span><span class="va">scoring</span><span class="op">$</span><span class="va">rsID</span> <span class="op">==</span> <span class="va">variant_id</span>, <span class="st">'effect_weight'</span>, drop <span class="op">=</span> <span class="cn">TRUE</span><span class="op">]</span>
  
  <span class="op">(</span><span class="va">alleles</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">==</span> <span class="va">eff</span><span class="op">)</span> <span class="op">*</span> <span class="va">weight</span> <span class="op">+</span> <span class="op">(</span><span class="va">alleles</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">==</span> <span class="va">eff</span><span class="op">)</span> <span class="op">*</span> <span class="va">weight</span>
<span class="op">}</span>, vectorize.args <span class="op">=</span> <span class="st">'genotype'</span><span class="op">)</span></code></pre></div>
<p>And, <em>voilà</em>, here are the risk scores for all possible genotypes:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">s</span> <span class="op">&lt;-</span> <span class="va">PRS_BrS</span><span class="op">$</span><span class="va">PGS000737</span><span class="op">$</span><span class="va">data</span>
<span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">BrS_genotypes</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>
    score <span class="op">=</span> <span class="fu">prs</span><span class="op">(</span><span class="va">s</span>, <span class="st">'rs11708996'</span>, <span class="va">rs11708996</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">prs</span><span class="op">(</span><span class="va">s</span>, <span class="st">'rs10428132'</span>, <span class="va">rs10428132</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu">prs</span><span class="op">(</span><span class="va">s</span>, <span class="st">'rs9388451'</span>, <span class="va">rs9388451</span><span class="op">)</span>
  <span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">scores</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 27 × 5</span>
<span class="co">#&gt;    patient rs11708996 rs10428132 rs9388451   score</span>
<span class="co">#&gt;      &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;       &lt;dbl&gt;</span>
<span class="co">#&gt;  1       1 G/G        T/T        T/T        0     </span>
<span class="co">#&gt;  2       2 G/G        T/T        T/C        0.46  </span>
<span class="co">#&gt;  3       3 G/G        T/T        C/C        0.92  </span>
<span class="co">#&gt;  4       4 G/G        T/G        T/T       -0.94  </span>
<span class="co">#&gt;  5       5 G/G        T/G        T/C       -0.48  </span>
<span class="co">#&gt;  6       6 G/G        T/G        C/C       -0.0200</span>
<span class="co">#&gt;  7       7 G/G        G/G        T/T       -1.88  </span>
<span class="co">#&gt;  8       8 G/G        G/G        T/C       -1.42  </span>
<span class="co">#&gt;  9       9 G/G        G/G        C/C       -0.96  </span>
<span class="co">#&gt; 10      10 G/C        T/T        T/T        0.55  </span>
<span class="co">#&gt; 11      11 G/C        T/T        T/C        1.01  </span>
<span class="co">#&gt; 12      12 G/C        T/T        C/C        1.47  </span>
<span class="co">#&gt; 13      13 G/C        T/G        T/T       -0.39  </span>
<span class="co">#&gt; 14      14 G/C        T/G        T/C        0.0700</span>
<span class="co">#&gt; 15      15 G/C        T/G        C/C        0.53  </span>
<span class="co">#&gt; 16      16 G/C        G/G        T/T       -1.33  </span>
<span class="co">#&gt; 17      17 G/C        G/G        T/C       -0.87  </span>
<span class="co">#&gt; 18      18 G/C        G/G        C/C       -0.410 </span>
<span class="co">#&gt; 19      19 C/C        T/T        T/T        1.1   </span>
<span class="co">#&gt; 20      20 C/C        T/T        T/C        1.56  </span>
<span class="co">#&gt; 21      21 C/C        T/T        C/C        2.02  </span>
<span class="co">#&gt; 22      22 C/C        T/G        T/T        0.160 </span>
<span class="co">#&gt; 23      23 C/C        T/G        T/C        0.62  </span>
<span class="co">#&gt; 24      24 C/C        T/G        C/C        1.08  </span>
<span class="co">#&gt; 25      25 C/C        G/G        T/T       -0.78  </span>
<span class="co">#&gt; 26      26 C/C        G/G        T/C       -0.320 </span>
<span class="co">#&gt; 27      27 C/C        G/G        C/C        0.140</span></code></pre></div>
<p>And below a graphical visualisation of the risk scores from lowest (top) to highest (bottom):</p>
<p><img src="risk-strat-brs-patients_files/figure-html/unnamed-chunk-13-1.svg" width="100%"></p>
<p>The polygenic risk score spans a range from (almost) <span class="math inline">\(-2\)</span> to (litle more than) <span class="math inline">\(2\)</span>. Those genotypes with a negative score carry mostly “protective” alleles, e.g., the most extreme case being <code>G/G-G/G-T/T</code> (containing only “protective” alleles)—individuals carrying this genotype have the lowest risk of developing the Brugada syndrome; on the other hand, individuals carrying the genotype <code>C/C-T/T-C/C</code> have the highest risk, and should probably be subject to closer clinical follow-up.</p>
</div>
<div id="final-remarks" class="section level2">
<h2 class="hasAnchor">
<a href="#final-remarks" class="anchor"></a>Final remarks</h2>
<p>The question of how to use the risk scores obtained, namely, how to proceed clinically, is an open and on-going discussion in the medical community that typically needs to be considered in the larger context of each individual characteristics. A good place to start looking into this discussion is the original paper which published this polygenic risk score: <a href="https://academic.oup.com/eurheartj/article/40/37/3097/5559542">https://academic.oup.com/eurheartj/article/40/37/3097/5559542</a>.</p>
<p>Furthermore, the purpose of this vignette is to show how to start from a trait or disease of interest, and end up by actually calculating the risk scores for particular individuals (if the user has real patient data). In this example, the individuals are hypothetical, and are represented by their in silico constructed genotypes. While trying to apply the same workflow to other traits or diseases, things are likely to be different from this simple example, namely, the number of variants will likely be much larger, and the polygenic risk score model applied could be more complex than the simple additive architecture assumed here. Also, you will probably want to take advantage of a dedicated tool that specialises in score computation instead of using your own custom R code, e.g., you could use the popular <a href="https://privefl.github.io/bigsnpr/articles/LDpred2.html">bigsnpr</a> package.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://rmagno.eu/">Ramiro Magno</a>, <a href="https://iduarte.eu/">Isabel Duarte</a>, Ana-Teresa Maia, Center for Research in Health Technologies and Information Systems (CINTESIS), University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '6eb557c1e1320049dc93c778f9efe639',
    indexName: 'quincunx',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
